// create-vue/bundle.js
#!/usr/bin/env node
import { createRequire } from "module";
import * as fs$5 from "node:fs";
import * as fs$4 from "node:fs";
import * as fs$3 from "node:fs";
import * as fs$2 from "node:fs";
import * as fs$1 from "node:fs";
import * as fs from "node:fs";
import * as path$5 from "node:path";
import * as path$4 from "node:path";
import * as path$3 from "node:path";
import * as path$2 from "node:path";
import * as path from "node:path";
import { fileURLToPath, pathToFileURL } from "node:url";
import { parseArgs, stripVTControlCharacters } from "node:util";
import g, { stdin, stdout } from "node:process";
import * as f$1 from "node:readline";
import M from "node:readline";
import { WriteStream } from "node:tty";
import * as path$1 from "path";

//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __commonJS = (cb, mod) => function() {
	return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k$2) => from[k$2]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));
var __require = /* @__PURE__ */ createRequire(import.meta.url);

//#endregion
//#region node_modules/.pnpm/sisteransi@1.0.5/node_modules/sisteransi/src/index.js
var require_src = __commonJS({ "node_modules/.pnpm/sisteransi@1.0.5/node_modules/sisteransi/src/index.js"(exports, module) {
	const ESC = "\x1B";
	const CSI = `${ESC}[`;
	const beep = "\x07";
	const cursor = {
		to(x$1, y$1) {
			if (!y$1) return `${CSI}${x$1 + 1}G`;
			return `${CSI}${y$1 + 1};${x$1 + 1}H`;
		},
		move(x$1, y$1) {
			let ret = "";
			if (x$1 < 0) ret += `${CSI}${-x$1}D`;
			else if (x$1 > 0) ret += `${CSI}${x$1}C`;
			if (y$1 < 0) ret += `${CSI}${-y$1}A`;
			else if (y$1 > 0) ret += `${CSI}${y$1}B`;
			return ret;
		},
		up: (count = 1) => `${CSI}${count}A`,
		down: (count = 1) => `${CSI}${count}B`,
		forward: (count = 1) => `${CSI}${count}C`,
		backward: (count = 1) => `${CSI}${count}D`,
		nextLine: (count = 1) => `${CSI}E`.repeat(count),
		prevLine: (count = 1) => `${CSI}F`.repeat(count),
		left: `${CSI}G`,
		hide: `${CSI}?25l`,
		show: `${CSI}?25h`,
		save: `${ESC}7`,
		restore: `${ESC}8`
	};
	const scroll = {
		up: (count = 1) => `${CSI}S`.repeat(count),
		down: (count = 1) => `${CSI}T`.repeat(count)
	};
	const erase = {
		screen: `${CSI}2J`,
		up: (count = 1) => `${CSI}1J`.repeat(count),
		down: (count = 1) => `${CSI}J`.repeat(count),
		line: `${CSI}2K`,
		lineEnd: `${CSI}K`,
		lineStart: `${CSI}1K`,
		lines(count) {
			let clear = "";
			for (let i = 0; i < count; i++) clear += this.line + (i < count - 1 ? cursor.up() : "");
			if (count) clear += cursor.left;
			return clear;
		}
	};
	module.exports = {
		cursor,
		scroll,
		erase,
		beep
	};
} });
var import_src = __toESM(require_src(), 1);

//#endregion
//#region node_modules/.pnpm/picocolors@1.1.1/node_modules/picocolors/picocolors.js
var require_picocolors = __commonJS({ "node_modules/.pnpm/picocolors@1.1.1/node_modules/picocolors/picocolors.js"(exports, module) {
	let p = process || {}, argv = p.argv || [], env = p.env || {};
	let isColorSupported = !(!!env.NO_COLOR || argv.includes("--no-color")) && (!!env.FORCE_COLOR || argv.includes("--color") || p.platform === "win32" || (p.stdout || {}).isTTY && env.TERM !== "dumb" || !!env.CI);
	let formatter = (open, close, replace = open) => (input) => {
		let string = "" + input, index = string.indexOf(close, open.length);
		return ~index ? open + replaceClose(string, close, replace, index) + close : open + string + close;
	};
	let replaceClose = (string, close, replace, index) => {
		let result = "", cursor$1 = 0;
		do {
			result += string.substring(cursor$1, index) + replace;
			cursor$1 = index + close.length;
			index = string.indexOf(close, cursor$1);
		} while (~index);
		return result + string.substring(cursor$1);
	};
	let createColors = (enabled = isColorSupported) => {
		let f$2 = enabled ? formatter : () => String;
		return {
			isColorSupported: enabled,
			reset: f$2("\x1B[0m", "\x1B[0m"),
			bold: f$2("\x1B[1m", "\x1B[22m", "\x1B[22m\x1B[1m"),
			dim: f$2("\x1B[2m", "\x1B[22m", "\x1B[22m\x1B[2m"),
			italic: f$2("\x1B[3m", "\x1B[23m"),
			underline: f$2("\x1B[4m", "\x1B[24m"),
			inverse: f$2("\x1B[7m", "\x1B[27m"),
			hidden: f$2("\x1B[8m", "\x1B[28m"),
			strikethrough: f$2("\x1B[9m", "\x1B[29m"),
			black: f$2("\x1B[30m", "\x1B[39m"),
			red: f$2("\x1B[31m", "\x1B[39m"),
			green: f$2("\x1B[32m", "\x1B[39m"),
			yellow: f$2("\x1B[33m", "\x1B[39m"),
			blue: f$2("\x1B[34m", "\x1B[39m"),
			magenta: f$2("\x1B[35m", "\x1B[39m"),
			cyan: f$2("\x1B[36m", "\x1B[39m"),
			white: f$2("\x1B[37m", "\x1B[39m"),
			gray: f$2("\x1B[90m", "\x1B[39m"),
			bgBlack: f$2("\x1B[40m", "\x1B[49m"),
			bgRed: f$2("\x1B[41m", "\x1B[49m"),
			bgGreen: f$2("\x1B[42m", "\x1B[49m"),
			bgYellow: f$2("\x1B[43m", "\x1B[49m"),
			bgBlue: f$2("\x1B[44m", "\x1B[49m"),
			bgMagenta: f$2("\x1B[45m", "\x1B[49m"),
			bgCyan: f$2("\x1B[46m", "\x1B[49m"),
			bgWhite: f$2("\x1B[47m", "\x1B[49m"),
			blackBright: f$2("\x1B[90m", "\x1B[39m"),
			redBright: f$2("\x1B[91m", "\x1B[39m"),
			greenBright: f$2("\x1B[92m", "\x1B[39m"),
			yellowBright: f$2("\x1B[93m", "\x1B[39m"),
			blueBright: f$2("\x1B[94m", "\x1B[39m"),
			magentaBright: f$2("\x1B[95m", "\x1B[39m"),
			cyanBright: f$2("\x1B[96m", "\x1B[39m"),
			whiteBright: f$2("\x1B[97m", "\x1B[39m"),
			bgBlackBright: f$2("\x1B[100m", "\x1B[49m"),
			bgRedBright: f$2("\x1B[101m", "\x1B[49m"),
			bgGreenBright: f$2("\x1B[102m", "\x1B[49m"),
			bgYellowBright: f$2("\x1B[103m", "\x1B[49m"),
			bgBlueBright: f$2("\x1B[104m", "\x1B[49m"),
			bgMagentaBright: f$2("\x1B[105m", "\x1B[49m"),
			bgCyanBright: f$2("\x1B[106m", "\x1B[49m"),
			bgWhiteBright: f$2("\x1B[107m", "\x1B[49m")
		};
	};
	module.exports = createColors();
	module.exports.createColors = createColors;
} });
var import_picocolors = __toESM(require_picocolors(), 1);

//#endregion
//#region node_modules/.pnpm/@clack+core@0.4.1/node_modules/@clack/core/dist/index.mjs
function J$1({ onlyFirst: t = !1 } = {}) {
	const F$1 = ["[\\u001B\\u009B][[\\]()#;?]*(?:(?:(?:(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]+)*|[a-zA-Z\\d]+(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]*)*)?(?:\\u0007|\\u001B\\u005C|\\u009C))", "(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PR-TZcf-nq-uy=><~]))"].join("|");
	return new RegExp(F$1, t ? void 0 : "g");
}
const Q = J$1();
function T$1(t) {
	if (typeof t != "string") throw new TypeError(`Expected a \`string\`, got \`${typeof t}\``);
	return t.replace(Q, "");
}
function O(t) {
	return t && t.__esModule && Object.prototype.hasOwnProperty.call(t, "default") ? t.default : t;
}
var P$1 = { exports: {} };
(function(t) {
	var u$1 = {};
	t.exports = u$1, u$1.eastAsianWidth = function(e) {
		var s = e.charCodeAt(0), i = e.length == 2 ? e.charCodeAt(1) : 0, D$1 = s;
		return 55296 <= s && s <= 56319 && 56320 <= i && i <= 57343 && (s &= 1023, i &= 1023, D$1 = s << 10 | i, D$1 += 65536), D$1 == 12288 || 65281 <= D$1 && D$1 <= 65376 || 65504 <= D$1 && D$1 <= 65510 ? "F" : D$1 == 8361 || 65377 <= D$1 && D$1 <= 65470 || 65474 <= D$1 && D$1 <= 65479 || 65482 <= D$1 && D$1 <= 65487 || 65490 <= D$1 && D$1 <= 65495 || 65498 <= D$1 && D$1 <= 65500 || 65512 <= D$1 && D$1 <= 65518 ? "H" : 4352 <= D$1 && D$1 <= 4447 || 4515 <= D$1 && D$1 <= 4519 || 4602 <= D$1 && D$1 <= 4607 || 9001 <= D$1 && D$1 <= 9002 || 11904 <= D$1 && D$1 <= 11929 || 11931 <= D$1 && D$1 <= 12019 || 12032 <= D$1 && D$1 <= 12245 || 12272 <= D$1 && D$1 <= 12283 || 12289 <= D$1 && D$1 <= 12350 || 12353 <= D$1 && D$1 <= 12438 || 12441 <= D$1 && D$1 <= 12543 || 12549 <= D$1 && D$1 <= 12589 || 12593 <= D$1 && D$1 <= 12686 || 12688 <= D$1 && D$1 <= 12730 || 12736 <= D$1 && D$1 <= 12771 || 12784 <= D$1 && D$1 <= 12830 || 12832 <= D$1 && D$1 <= 12871 || 12880 <= D$1 && D$1 <= 13054 || 13056 <= D$1 && D$1 <= 19903 || 19968 <= D$1 && D$1 <= 42124 || 42128 <= D$1 && D$1 <= 42182 || 43360 <= D$1 && D$1 <= 43388 || 44032 <= D$1 && D$1 <= 55203 || 55216 <= D$1 && D$1 <= 55238 || 55243 <= D$1 && D$1 <= 55291 || 63744 <= D$1 && D$1 <= 64255 || 65040 <= D$1 && D$1 <= 65049 || 65072 <= D$1 && D$1 <= 65106 || 65108 <= D$1 && D$1 <= 65126 || 65128 <= D$1 && D$1 <= 65131 || 110592 <= D$1 && D$1 <= 110593 || 127488 <= D$1 && D$1 <= 127490 || 127504 <= D$1 && D$1 <= 127546 || 127552 <= D$1 && D$1 <= 127560 || 127568 <= D$1 && D$1 <= 127569 || 131072 <= D$1 && D$1 <= 194367 || 177984 <= D$1 && D$1 <= 196605 || 196608 <= D$1 && D$1 <= 262141 ? "W" : 32 <= D$1 && D$1 <= 126 || 162 <= D$1 && D$1 <= 163 || 165 <= D$1 && D$1 <= 166 || D$1 == 172 || D$1 == 175 || 10214 <= D$1 && D$1 <= 10221 || 10629 <= D$1 && D$1 <= 10630 ? "Na" : D$1 == 161 || D$1 == 164 || 167 <= D$1 && D$1 <= 168 || D$1 == 170 || 173 <= D$1 && D$1 <= 174 || 176 <= D$1 && D$1 <= 180 || 182 <= D$1 && D$1 <= 186 || 188 <= D$1 && D$1 <= 191 || D$1 == 198 || D$1 == 208 || 215 <= D$1 && D$1 <= 216 || 222 <= D$1 && D$1 <= 225 || D$1 == 230 || 232 <= D$1 && D$1 <= 234 || 236 <= D$1 && D$1 <= 237 || D$1 == 240 || 242 <= D$1 && D$1 <= 243 || 247 <= D$1 && D$1 <= 250 || D$1 == 252 || D$1 == 254 || D$1 == 257 || D$1 == 273 || D$1 == 275 || D$1 == 283 || 294 <= D$1 && D$1 <= 295 || D$1 == 299 || 305 <= D$1 && D$1 <= 307 || D$1 == 312 || 319 <= D$1 && D$1 <= 322 || D$1 == 324 || 328 <= D$1 && D$1 <= 331 || D$1 == 333 || 338 <= D$1 && D$1 <= 339 || 358 <= D$1 && D$1 <= 359 || D$1 == 363 || D$1 == 462 || D$1 == 464 || D$1 == 466 || D$1 == 468 || D$1 == 470 || D$1 == 472 || D$1 == 474 || D$1 == 476 || D$1 == 593 || D$1 == 609 || D$1 == 708 || D$1 == 711 || 713 <= D$1 && D$1 <= 715 || D$1 == 717 || D$1 == 720 || 728 <= D$1 && D$1 <= 731 || D$1 == 733 || D$1 == 735 || 768 <= D$1 && D$1 <= 879 || 913 <= D$1 && D$1 <= 929 || 931 <= D$1 && D$1 <= 937 || 945 <= D$1 && D$1 <= 961 || 963 <= D$1 && D$1 <= 969 || D$1 == 1025 || 1040 <= D$1 && D$1 <= 1103 || D$1 == 1105 || D$1 == 8208 || 8211 <= D$1 && D$1 <= 8214 || 8216 <= D$1 && D$1 <= 8217 || 8220 <= D$1 && D$1 <= 8221 || 8224 <= D$1 && D$1 <= 8226 || 8228 <= D$1 && D$1 <= 8231 || D$1 == 8240 || 8242 <= D$1 && D$1 <= 8243 || D$1 == 8245 || D$1 == 8251 || D$1 == 8254 || D$1 == 8308 || D$1 == 8319 || 8321 <= D$1 && D$1 <= 8324 || D$1 == 8364 || D$1 == 8451 || D$1 == 8453 || D$1 == 8457 || D$1 == 8467 || D$1 == 8470 || 8481 <= D$1 && D$1 <= 8482 || D$1 == 8486 || D$1 == 8491 || 8531 <= D$1 && D$1 <= 8532 || 8539 <= D$1 && D$1 <= 8542 || 8544 <= D$1 && D$1 <= 8555 || 8560 <= D$1 && D$1 <= 8569 || D$1 == 8585 || 8592 <= D$1 && D$1 <= 8601 || 8632 <= D$1 && D$1 <= 8633 || D$1 == 8658 || D$1 == 8660 || D$1 == 8679 || D$1 == 8704 || 8706 <= D$1 && D$1 <= 8707 || 8711 <= D$1 && D$1 <= 8712 || D$1 == 8715 || D$1 == 8719 || D$1 == 8721 || D$1 == 8725 || D$1 == 8730 || 8733 <= D$1 && D$1 <= 8736 || D$1 == 8739 || D$1 == 8741 || 8743 <= D$1 && D$1 <= 8748 || D$1 == 8750 || 8756 <= D$1 && D$1 <= 8759 || 8764 <= D$1 && D$1 <= 8765 || D$1 == 8776 || D$1 == 8780 || D$1 == 8786 || 8800 <= D$1 && D$1 <= 8801 || 8804 <= D$1 && D$1 <= 8807 || 8810 <= D$1 && D$1 <= 8811 || 8814 <= D$1 && D$1 <= 8815 || 8834 <= D$1 && D$1 <= 8835 || 8838 <= D$1 && D$1 <= 8839 || D$1 == 8853 || D$1 == 8857 || D$1 == 8869 || D$1 == 8895 || D$1 == 8978 || 9312 <= D$1 && D$1 <= 9449 || 9451 <= D$1 && D$1 <= 9547 || 9552 <= D$1 && D$1 <= 9587 || 9600 <= D$1 && D$1 <= 9615 || 9618 <= D$1 && D$1 <= 9621 || 9632 <= D$1 && D$1 <= 9633 || 9635 <= D$1 && D$1 <= 9641 || 9650 <= D$1 && D$1 <= 9651 || 9654 <= D$1 && D$1 <= 9655 || 9660 <= D$1 && D$1 <= 9661 || 9664 <= D$1 && D$1 <= 9665 || 9670 <= D$1 && D$1 <= 9672 || D$1 == 9675 || 9678 <= D$1 && D$1 <= 9681 || 9698 <= D$1 && D$1 <= 9701 || D$1 == 9711 || 9733 <= D$1 && D$1 <= 9734 || D$1 == 9737 || 9742 <= D$1 && D$1 <= 9743 || 9748 <= D$1 && D$1 <= 9749 || D$1 == 9756 || D$1 == 9758 || D$1 == 9792 || D$1 == 9794 || 9824 <= D$1 && D$1 <= 9825 || 9827 <= D$1 && D$1 <= 9829 || 9831 <= D$1 && D$1 <= 9834 || 9836 <= D$1 && D$1 <= 9837 || D$1 == 9839 || 9886 <= D$1 && D$1 <= 9887 || 9918 <= D$1 && D$1 <= 9919 || 9924 <= D$1 && D$1 <= 9933 || 9935 <= D$1 && D$1 <= 9953 || D$1 == 9955 || 9960 <= D$1 && D$1 <= 9983 || D$1 == 10045 || D$1 == 10071 || 10102 <= D$1 && D$1 <= 10111 || 11093 <= D$1 && D$1 <= 11097 || 12872 <= D$1 && D$1 <= 12879 || 57344 <= D$1 && D$1 <= 63743 || 65024 <= D$1 && D$1 <= 65039 || D$1 == 65533 || 127232 <= D$1 && D$1 <= 127242 || 127248 <= D$1 && D$1 <= 127277 || 127280 <= D$1 && D$1 <= 127337 || 127344 <= D$1 && D$1 <= 127386 || 917760 <= D$1 && D$1 <= 917999 || 983040 <= D$1 && D$1 <= 1048573 || 1048576 <= D$1 && D$1 <= 1114109 ? "A" : "N";
	}, u$1.characterLength = function(e) {
		var s = this.eastAsianWidth(e);
		return s == "F" || s == "W" || s == "A" ? 2 : 1;
	};
	function F$1(e) {
		return e.match(/[\uD800-\uDBFF][\uDC00-\uDFFF]|[^\uD800-\uDFFF]/g) || [];
	}
	u$1.length = function(e) {
		for (var s = F$1(e), i = 0, D$1 = 0; D$1 < s.length; D$1++) i = i + this.characterLength(s[D$1]);
		return i;
	}, u$1.slice = function(e, s, i) {
		textLen = u$1.length(e), s = s || 0, i = i || 1, s < 0 && (s = textLen + s), i < 0 && (i = textLen + i);
		for (var D$1 = "", C$1 = 0, o$1 = F$1(e), E = 0; E < o$1.length; E++) {
			var a = o$1[E], n = u$1.length(a);
			if (C$1 >= s - (n == 2 ? 1 : 0)) if (C$1 + n <= i) D$1 += a;
			else break;
			C$1 += n;
		}
		return D$1;
	};
})(P$1);
var X = P$1.exports;
const DD = O(X);
var uD = function() {
	return /\uD83C\uDFF4\uDB40\uDC67\uDB40\uDC62(?:\uDB40\uDC77\uDB40\uDC6C\uDB40\uDC73|\uDB40\uDC73\uDB40\uDC63\uDB40\uDC74|\uDB40\uDC65\uDB40\uDC6E\uDB40\uDC67)\uDB40\uDC7F|(?:\uD83E\uDDD1\uD83C\uDFFF\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83E\uDDD1|\uD83D\uDC69\uD83C\uDFFF\u200D\uD83E\uDD1D\u200D(?:\uD83D[\uDC68\uDC69]))(?:\uD83C[\uDFFB-\uDFFE])|(?:\uD83E\uDDD1\uD83C\uDFFE\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83E\uDDD1|\uD83D\uDC69\uD83C\uDFFE\u200D\uD83E\uDD1D\u200D(?:\uD83D[\uDC68\uDC69]))(?:\uD83C[\uDFFB-\uDFFD\uDFFF])|(?:\uD83E\uDDD1\uD83C\uDFFD\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83E\uDDD1|\uD83D\uDC69\uD83C\uDFFD\u200D\uD83E\uDD1D\u200D(?:\uD83D[\uDC68\uDC69]))(?:\uD83C[\uDFFB\uDFFC\uDFFE\uDFFF])|(?:\uD83E\uDDD1\uD83C\uDFFC\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83E\uDDD1|\uD83D\uDC69\uD83C\uDFFC\u200D\uD83E\uDD1D\u200D(?:\uD83D[\uDC68\uDC69]))(?:\uD83C[\uDFFB\uDFFD-\uDFFF])|(?:\uD83E\uDDD1\uD83C\uDFFB\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83E\uDDD1|\uD83D\uDC69\uD83C\uDFFB\u200D\uD83E\uDD1D\u200D(?:\uD83D[\uDC68\uDC69]))(?:\uD83C[\uDFFC-\uDFFF])|\uD83D\uDC68(?:\uD83C\uDFFB(?:\u200D(?:\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFF])|\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFF]))|\uD83E\uDD1D\u200D\uD83D\uDC68(?:\uD83C[\uDFFC-\uDFFF])|[\u2695\u2696\u2708]\uFE0F|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD]))?|(?:\uD83C[\uDFFC-\uDFFF])\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFF])|\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFF]))|\u200D(?:\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D)?\uD83D\uDC68|(?:\uD83D[\uDC68\uDC69])\u200D(?:\uD83D\uDC66\u200D\uD83D\uDC66|\uD83D\uDC67\u200D(?:\uD83D[\uDC66\uDC67]))|\uD83D\uDC66\u200D\uD83D\uDC66|\uD83D\uDC67\u200D(?:\uD83D[\uDC66\uDC67])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFF\u200D(?:\uD83E\uDD1D\u200D\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFE])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFE\u200D(?:\uD83E\uDD1D\u200D\uD83D\uDC68(?:\uD83C[\uDFFB-\uDFFD\uDFFF])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFD\u200D(?:\uD83E\uDD1D\u200D\uD83D\uDC68(?:\uD83C[\uDFFB\uDFFC\uDFFE\uDFFF])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFC\u200D(?:\uD83E\uDD1D\u200D\uD83D\uDC68(?:\uD83C[\uDFFB\uDFFD-\uDFFF])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|(?:\uD83C\uDFFF\u200D[\u2695\u2696\u2708]|\uD83C\uDFFE\u200D[\u2695\u2696\u2708]|\uD83C\uDFFD\u200D[\u2695\u2696\u2708]|\uD83C\uDFFC\u200D[\u2695\u2696\u2708]|\u200D[\u2695\u2696\u2708])\uFE0F|\u200D(?:(?:\uD83D[\uDC68\uDC69])\u200D(?:\uD83D[\uDC66\uDC67])|\uD83D[\uDC66\uDC67])|\uD83C\uDFFF|\uD83C\uDFFE|\uD83C\uDFFD|\uD83C\uDFFC)?|(?:\uD83D\uDC69(?:\uD83C\uDFFB\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D(?:\uD83D[\uDC68\uDC69])|\uD83D[\uDC68\uDC69])|(?:\uD83C[\uDFFC-\uDFFF])\u200D\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D(?:\uD83D[\uDC68\uDC69])|\uD83D[\uDC68\uDC69]))|\uD83E\uDDD1(?:\uD83C[\uDFFB-\uDFFF])\u200D\uD83E\uDD1D\u200D\uD83E\uDDD1)(?:\uD83C[\uDFFB-\uDFFF])|\uD83D\uDC69\u200D\uD83D\uDC69\u200D(?:\uD83D\uDC66\u200D\uD83D\uDC66|\uD83D\uDC67\u200D(?:\uD83D[\uDC66\uDC67]))|\uD83D\uDC69(?:\u200D(?:\u2764\uFE0F\u200D(?:\uD83D\uDC8B\u200D(?:\uD83D[\uDC68\uDC69])|\uD83D[\uDC68\uDC69])|\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFF\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFE\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFD\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFC\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFB\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD]))|\uD83E\uDDD1(?:\u200D(?:\uD83E\uDD1D\u200D\uD83E\uDDD1|\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFF\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFE\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFD\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFC\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD])|\uD83C\uDFFB\u200D(?:\uD83C[\uDF3E\uDF73\uDF7C\uDF84\uDF93\uDFA4\uDFA8\uDFEB\uDFED]|\uD83D[\uDCBB\uDCBC\uDD27\uDD2C\uDE80\uDE92]|\uD83E[\uDDAF-\uDDB3\uDDBC\uDDBD]))|\uD83D\uDC69\u200D\uD83D\uDC66\u200D\uD83D\uDC66|\uD83D\uDC69\u200D\uD83D\uDC69\u200D(?:\uD83D[\uDC66\uDC67])|\uD83D\uDC69\u200D\uD83D\uDC67\u200D(?:\uD83D[\uDC66\uDC67])|(?:\uD83D\uDC41\uFE0F\u200D\uD83D\uDDE8|\uD83E\uDDD1(?:\uD83C\uDFFF\u200D[\u2695\u2696\u2708]|\uD83C\uDFFE\u200D[\u2695\u2696\u2708]|\uD83C\uDFFD\u200D[\u2695\u2696\u2708]|\uD83C\uDFFC\u200D[\u2695\u2696\u2708]|\uD83C\uDFFB\u200D[\u2695\u2696\u2708]|\u200D[\u2695\u2696\u2708])|\uD83D\uDC69(?:\uD83C\uDFFF\u200D[\u2695\u2696\u2708]|\uD83C\uDFFE\u200D[\u2695\u2696\u2708]|\uD83C\uDFFD\u200D[\u2695\u2696\u2708]|\uD83C\uDFFC\u200D[\u2695\u2696\u2708]|\uD83C\uDFFB\u200D[\u2695\u2696\u2708]|\u200D[\u2695\u2696\u2708])|\uD83D\uDE36\u200D\uD83C\uDF2B|\uD83C\uDFF3\uFE0F\u200D\u26A7|\uD83D\uDC3B\u200D\u2744|(?:(?:\uD83C[\uDFC3\uDFC4\uDFCA]|\uD83D[\uDC6E\uDC70\uDC71\uDC73\uDC77\uDC81\uDC82\uDC86\uDC87\uDE45-\uDE47\uDE4B\uDE4D\uDE4E\uDEA3\uDEB4-\uDEB6]|\uD83E[\uDD26\uDD35\uDD37-\uDD39\uDD3D\uDD3E\uDDB8\uDDB9\uDDCD-\uDDCF\uDDD4\uDDD6-\uDDDD])(?:\uD83C[\uDFFB-\uDFFF])|\uD83D\uDC6F|\uD83E[\uDD3C\uDDDE\uDDDF])\u200D[\u2640\u2642]|(?:\u26F9|\uD83C[\uDFCB\uDFCC]|\uD83D\uDD75)(?:\uFE0F|\uD83C[\uDFFB-\uDFFF])\u200D[\u2640\u2642]|\uD83C\uDFF4\u200D\u2620|(?:\uD83C[\uDFC3\uDFC4\uDFCA]|\uD83D[\uDC6E\uDC70\uDC71\uDC73\uDC77\uDC81\uDC82\uDC86\uDC87\uDE45-\uDE47\uDE4B\uDE4D\uDE4E\uDEA3\uDEB4-\uDEB6]|\uD83E[\uDD26\uDD35\uDD37-\uDD39\uDD3D\uDD3E\uDDB8\uDDB9\uDDCD-\uDDCF\uDDD4\uDDD6-\uDDDD])\u200D[\u2640\u2642]|[\xA9\xAE\u203C\u2049\u2122\u2139\u2194-\u2199\u21A9\u21AA\u2328\u23CF\u23ED-\u23EF\u23F1\u23F2\u23F8-\u23FA\u24C2\u25AA\u25AB\u25B6\u25C0\u25FB\u25FC\u2600-\u2604\u260E\u2611\u2618\u2620\u2622\u2623\u2626\u262A\u262E\u262F\u2638-\u263A\u2640\u2642\u265F\u2660\u2663\u2665\u2666\u2668\u267B\u267E\u2692\u2694-\u2697\u2699\u269B\u269C\u26A0\u26A7\u26B0\u26B1\u26C8\u26CF\u26D1\u26D3\u26E9\u26F0\u26F1\u26F4\u26F7\u26F8\u2702\u2708\u2709\u270F\u2712\u2714\u2716\u271D\u2721\u2733\u2734\u2744\u2747\u2763\u27A1\u2934\u2935\u2B05-\u2B07\u3030\u303D\u3297\u3299]|\uD83C[\uDD70\uDD71\uDD7E\uDD7F\uDE02\uDE37\uDF21\uDF24-\uDF2C\uDF36\uDF7D\uDF96\uDF97\uDF99-\uDF9B\uDF9E\uDF9F\uDFCD\uDFCE\uDFD4-\uDFDF\uDFF5\uDFF7]|\uD83D[\uDC3F\uDCFD\uDD49\uDD4A\uDD6F\uDD70\uDD73\uDD76-\uDD79\uDD87\uDD8A-\uDD8D\uDDA5\uDDA8\uDDB1\uDDB2\uDDBC\uDDC2-\uDDC4\uDDD1-\uDDD3\uDDDC-\uDDDE\uDDE1\uDDE3\uDDE8\uDDEF\uDDF3\uDDFA\uDECB\uDECD-\uDECF\uDEE0-\uDEE5\uDEE9\uDEF0\uDEF3])\uFE0F|\uD83C\uDFF3\uFE0F\u200D\uD83C\uDF08|\uD83D\uDC69\u200D\uD83D\uDC67|\uD83D\uDC69\u200D\uD83D\uDC66|\uD83D\uDE35\u200D\uD83D\uDCAB|\uD83D\uDE2E\u200D\uD83D\uDCA8|\uD83D\uDC15\u200D\uD83E\uDDBA|\uD83E\uDDD1(?:\uD83C\uDFFF|\uD83C\uDFFE|\uD83C\uDFFD|\uD83C\uDFFC|\uD83C\uDFFB)?|\uD83D\uDC69(?:\uD83C\uDFFF|\uD83C\uDFFE|\uD83C\uDFFD|\uD83C\uDFFC|\uD83C\uDFFB)?|\uD83C\uDDFD\uD83C\uDDF0|\uD83C\uDDF6\uD83C\uDDE6|\uD83C\uDDF4\uD83C\uDDF2|\uD83D\uDC08\u200D\u2B1B|\u2764\uFE0F\u200D(?:\uD83D\uDD25|\uD83E\uDE79)|\uD83D\uDC41\uFE0F|\uD83C\uDFF3\uFE0F|\uD83C\uDDFF(?:\uD83C[\uDDE6\uDDF2\uDDFC])|\uD83C\uDDFE(?:\uD83C[\uDDEA\uDDF9])|\uD83C\uDDFC(?:\uD83C[\uDDEB\uDDF8])|\uD83C\uDDFB(?:\uD83C[\uDDE6\uDDE8\uDDEA\uDDEC\uDDEE\uDDF3\uDDFA])|\uD83C\uDDFA(?:\uD83C[\uDDE6\uDDEC\uDDF2\uDDF3\uDDF8\uDDFE\uDDFF])|\uD83C\uDDF9(?:\uD83C[\uDDE6\uDDE8\uDDE9\uDDEB-\uDDED\uDDEF-\uDDF4\uDDF7\uDDF9\uDDFB\uDDFC\uDDFF])|\uD83C\uDDF8(?:\uD83C[\uDDE6-\uDDEA\uDDEC-\uDDF4\uDDF7-\uDDF9\uDDFB\uDDFD-\uDDFF])|\uD83C\uDDF7(?:\uD83C[\uDDEA\uDDF4\uDDF8\uDDFA\uDDFC])|\uD83C\uDDF5(?:\uD83C[\uDDE6\uDDEA-\uDDED\uDDF0-\uDDF3\uDDF7-\uDDF9\uDDFC\uDDFE])|\uD83C\uDDF3(?:\uD83C[\uDDE6\uDDE8\uDDEA-\uDDEC\uDDEE\uDDF1\uDDF4\uDDF5\uDDF7\uDDFA\uDDFF])|\uD83C\uDDF2(?:\uD83C[\uDDE6\uDDE8-\uDDED\uDDF0-\uDDFF])|\uD83C\uDDF1(?:\uD83C[\uDDE6-\uDDE8\uDDEE\uDDF0\uDDF7-\uDDFB\uDDFE])|\uD83C\uDDF0(?:\uD83C[\uDDEA\uDDEC-\uDDEE\uDDF2\uDDF3\uDDF5\uDDF7\uDDFC\uDDFE\uDDFF])|\uD83C\uDDEF(?:\uD83C[\uDDEA\uDDF2\uDDF4\uDDF5])|\uD83C\uDDEE(?:\uD83C[\uDDE8-\uDDEA\uDDF1-\uDDF4\uDDF6-\uDDF9])|\uD83C\uDDED(?:\uD83C[\uDDF0\uDDF2\uDDF3\uDDF7\uDDF9\uDDFA])|\uD83C\uDDEC(?:\uD83C[\uDDE6\uDDE7\uDDE9-\uDDEE\uDDF1-\uDDF3\uDDF5-\uDDFA\uDDFC\uDDFE])|\uD83C\uDDEB(?:\uD83C[\uDDEE-\uDDF0\uDDF2\uDDF4\uDDF7])|\uD83C\uDDEA(?:\uD83C[\uDDE6\uDDE8\uDDEA\uDDEC\uDDED\uDDF7-\uDDFA])|\uD83C\uDDE9(?:\uD83C[\uDDEA\uDDEC\uDDEF\uDDF0\uDDF2\uDDF4\uDDFF])|\uD83C\uDDE8(?:\uD83C[\uDDE6\uDDE8\uDDE9\uDDEB-\uDDEE\uDDF0-\uDDF5\uDDF7\uDDFA-\uDDFF])|\uD83C\uDDE7(?:\uD83C[\uDDE6\uDDE7\uDDE9-\uDDEF\uDDF1-\uDDF4\uDDF6-\uDDF9\uDDFB\uDDFC\uDDFE\uDDFF])|\uD83C\uDDE6(?:\uD83C[\uDDE8-\uDDEC\uDDEE\uDDF1\uDDF2\uDDF4\uDDF6-\uDDFA\uDDFC\uDDFD\uDDFF])|[#\*0-9]\uFE0F\u20E3|\u2764\uFE0F|(?:\uD83C[\uDFC3\uDFC4\uDFCA]|\uD83D[\uDC6E\uDC70\uDC71\uDC73\uDC77\uDC81\uDC82\uDC86\uDC87\uDE45-\uDE47\uDE4B\uDE4D\uDE4E\uDEA3\uDEB4-\uDEB6]|\uD83E[\uDD26\uDD35\uDD37-\uDD39\uDD3D\uDD3E\uDDB8\uDDB9\uDDCD-\uDDCF\uDDD4\uDDD6-\uDDDD])(?:\uD83C[\uDFFB-\uDFFF])|(?:\u26F9|\uD83C[\uDFCB\uDFCC]|\uD83D\uDD75)(?:\uFE0F|\uD83C[\uDFFB-\uDFFF])|\uD83C\uDFF4|(?:[\u270A\u270B]|\uD83C[\uDF85\uDFC2\uDFC7]|\uD83D[\uDC42\uDC43\uDC46-\uDC50\uDC66\uDC67\uDC6B-\uDC6D\uDC72\uDC74-\uDC76\uDC78\uDC7C\uDC83\uDC85\uDC8F\uDC91\uDCAA\uDD7A\uDD95\uDD96\uDE4C\uDE4F\uDEC0\uDECC]|\uD83E[\uDD0C\uDD0F\uDD18-\uDD1C\uDD1E\uDD1F\uDD30-\uDD34\uDD36\uDD77\uDDB5\uDDB6\uDDBB\uDDD2\uDDD3\uDDD5])(?:\uD83C[\uDFFB-\uDFFF])|(?:[\u261D\u270C\u270D]|\uD83D[\uDD74\uDD90])(?:\uFE0F|\uD83C[\uDFFB-\uDFFF])|[\u270A\u270B]|\uD83C[\uDF85\uDFC2\uDFC7]|\uD83D[\uDC08\uDC15\uDC3B\uDC42\uDC43\uDC46-\uDC50\uDC66\uDC67\uDC6B-\uDC6D\uDC72\uDC74-\uDC76\uDC78\uDC7C\uDC83\uDC85\uDC8F\uDC91\uDCAA\uDD7A\uDD95\uDD96\uDE2E\uDE35\uDE36\uDE4C\uDE4F\uDEC0\uDECC]|\uD83E[\uDD0C\uDD0F\uDD18-\uDD1C\uDD1E\uDD1F\uDD30-\uDD34\uDD36\uDD77\uDDB5\uDDB6\uDDBB\uDDD2\uDDD3\uDDD5]|\uD83C[\uDFC3\uDFC4\uDFCA]|\uD83D[\uDC6E\uDC70\uDC71\uDC73\uDC77\uDC81\uDC82\uDC86\uDC87\uDE45-\uDE47\uDE4B\uDE4D\uDE4E\uDEA3\uDEB4-\uDEB6]|\uD83E[\uDD26\uDD35\uDD37-\uDD39\uDD3D\uDD3E\uDDB8\uDDB9\uDDCD-\uDDCF\uDDD4\uDDD6-\uDDDD]|\uD83D\uDC6F|\uD83E[\uDD3C\uDDDE\uDDDF]|[\u231A\u231B\u23E9-\u23EC\u23F0\u23F3\u25FD\u25FE\u2614\u2615\u2648-\u2653\u267F\u2693\u26A1\u26AA\u26AB\u26BD\u26BE\u26C4\u26C5\u26CE\u26D4\u26EA\u26F2\u26F3\u26F5\u26FA\u26FD\u2705\u2728\u274C\u274E\u2753-\u2755\u2757\u2795-\u2797\u27B0\u27BF\u2B1B\u2B1C\u2B50\u2B55]|\uD83C[\uDC04\uDCCF\uDD8E\uDD91-\uDD9A\uDE01\uDE1A\uDE2F\uDE32-\uDE36\uDE38-\uDE3A\uDE50\uDE51\uDF00-\uDF20\uDF2D-\uDF35\uDF37-\uDF7C\uDF7E-\uDF84\uDF86-\uDF93\uDFA0-\uDFC1\uDFC5\uDFC6\uDFC8\uDFC9\uDFCF-\uDFD3\uDFE0-\uDFF0\uDFF8-\uDFFF]|\uD83D[\uDC00-\uDC07\uDC09-\uDC14\uDC16-\uDC3A\uDC3C-\uDC3E\uDC40\uDC44\uDC45\uDC51-\uDC65\uDC6A\uDC79-\uDC7B\uDC7D-\uDC80\uDC84\uDC88-\uDC8E\uDC90\uDC92-\uDCA9\uDCAB-\uDCFC\uDCFF-\uDD3D\uDD4B-\uDD4E\uDD50-\uDD67\uDDA4\uDDFB-\uDE2D\uDE2F-\uDE34\uDE37-\uDE44\uDE48-\uDE4A\uDE80-\uDEA2\uDEA4-\uDEB3\uDEB7-\uDEBF\uDEC1-\uDEC5\uDED0-\uDED2\uDED5-\uDED7\uDEEB\uDEEC\uDEF4-\uDEFC\uDFE0-\uDFEB]|\uD83E[\uDD0D\uDD0E\uDD10-\uDD17\uDD1D\uDD20-\uDD25\uDD27-\uDD2F\uDD3A\uDD3F-\uDD45\uDD47-\uDD76\uDD78\uDD7A-\uDDB4\uDDB7\uDDBA\uDDBC-\uDDCB\uDDD0\uDDE0-\uDDFF\uDE70-\uDE74\uDE78-\uDE7A\uDE80-\uDE86\uDE90-\uDEA8\uDEB0-\uDEB6\uDEC0-\uDEC2\uDED0-\uDED6]|(?:[\u231A\u231B\u23E9-\u23EC\u23F0\u23F3\u25FD\u25FE\u2614\u2615\u2648-\u2653\u267F\u2693\u26A1\u26AA\u26AB\u26BD\u26BE\u26C4\u26C5\u26CE\u26D4\u26EA\u26F2\u26F3\u26F5\u26FA\u26FD\u2705\u270A\u270B\u2728\u274C\u274E\u2753-\u2755\u2757\u2795-\u2797\u27B0\u27BF\u2B1B\u2B1C\u2B50\u2B55]|\uD83C[\uDC04\uDCCF\uDD8E\uDD91-\uDD9A\uDDE6-\uDDFF\uDE01\uDE1A\uDE2F\uDE32-\uDE36\uDE38-\uDE3A\uDE50\uDE51\uDF00-\uDF20\uDF2D-\uDF35\uDF37-\uDF7C\uDF7E-\uDF93\uDFA0-\uDFCA\uDFCF-\uDFD3\uDFE0-\uDFF0\uDFF4\uDFF8-\uDFFF]|\uD83D[\uDC00-\uDC3E\uDC40\uDC42-\uDCFC\uDCFF-\uDD3D\uDD4B-\uDD4E\uDD50-\uDD67\uDD7A\uDD95\uDD96\uDDA4\uDDFB-\uDE4F\uDE80-\uDEC5\uDECC\uDED0-\uDED2\uDED5-\uDED7\uDEEB\uDEEC\uDEF4-\uDEFC\uDFE0-\uDFEB]|\uD83E[\uDD0C-\uDD3A\uDD3C-\uDD45\uDD47-\uDD78\uDD7A-\uDDCB\uDDCD-\uDDFF\uDE70-\uDE74\uDE78-\uDE7A\uDE80-\uDE86\uDE90-\uDEA8\uDEB0-\uDEB6\uDEC0-\uDEC2\uDED0-\uDED6])|(?:[#\*0-9\xA9\xAE\u203C\u2049\u2122\u2139\u2194-\u2199\u21A9\u21AA\u231A\u231B\u2328\u23CF\u23E9-\u23F3\u23F8-\u23FA\u24C2\u25AA\u25AB\u25B6\u25C0\u25FB-\u25FE\u2600-\u2604\u260E\u2611\u2614\u2615\u2618\u261D\u2620\u2622\u2623\u2626\u262A\u262E\u262F\u2638-\u263A\u2640\u2642\u2648-\u2653\u265F\u2660\u2663\u2665\u2666\u2668\u267B\u267E\u267F\u2692-\u2697\u2699\u269B\u269C\u26A0\u26A1\u26A7\u26AA\u26AB\u26B0\u26B1\u26BD\u26BE\u26C4\u26C5\u26C8\u26CE\u26CF\u26D1\u26D3\u26D4\u26E9\u26EA\u26F0-\u26F5\u26F7-\u26FA\u26FD\u2702\u2705\u2708-\u270D\u270F\u2712\u2714\u2716\u271D\u2721\u2728\u2733\u2734\u2744\u2747\u274C\u274E\u2753-\u2755\u2757\u2763\u2764\u2795-\u2797\u27A1\u27B0\u27BF\u2934\u2935\u2B05-\u2B07\u2B1B\u2B1C\u2B50\u2B55\u3030\u303D\u3297\u3299]|\uD83C[\uDC04\uDCCF\uDD70\uDD71\uDD7E\uDD7F\uDD8E\uDD91-\uDD9A\uDDE6-\uDDFF\uDE01\uDE02\uDE1A\uDE2F\uDE32-\uDE3A\uDE50\uDE51\uDF00-\uDF21\uDF24-\uDF93\uDF96\uDF97\uDF99-\uDF9B\uDF9E-\uDFF0\uDFF3-\uDFF5\uDFF7-\uDFFF]|\uD83D[\uDC00-\uDCFD\uDCFF-\uDD3D\uDD49-\uDD4E\uDD50-\uDD67\uDD6F\uDD70\uDD73-\uDD7A\uDD87\uDD8A-\uDD8D\uDD90\uDD95\uDD96\uDDA4\uDDA5\uDDA8\uDDB1\uDDB2\uDDBC\uDDC2-\uDDC4\uDDD1-\uDDD3\uDDDC-\uDDDE\uDDE1\uDDE3\uDDE8\uDDEF\uDDF3\uDDFA-\uDE4F\uDE80-\uDEC5\uDECB-\uDED2\uDED5-\uDED7\uDEE0-\uDEE5\uDEE9\uDEEB\uDEEC\uDEF0\uDEF3-\uDEFC\uDFE0-\uDFEB]|\uD83E[\uDD0C-\uDD3A\uDD3C-\uDD45\uDD47-\uDD78\uDD7A-\uDDCB\uDDCD-\uDDFF\uDE70-\uDE74\uDE78-\uDE7A\uDE80-\uDE86\uDE90-\uDEA8\uDEB0-\uDEB6\uDEC0-\uDEC2\uDED0-\uDED6])\uFE0F|(?:[\u261D\u26F9\u270A-\u270D]|\uD83C[\uDF85\uDFC2-\uDFC4\uDFC7\uDFCA-\uDFCC]|\uD83D[\uDC42\uDC43\uDC46-\uDC50\uDC66-\uDC78\uDC7C\uDC81-\uDC83\uDC85-\uDC87\uDC8F\uDC91\uDCAA\uDD74\uDD75\uDD7A\uDD90\uDD95\uDD96\uDE45-\uDE47\uDE4B-\uDE4F\uDEA3\uDEB4-\uDEB6\uDEC0\uDECC]|\uD83E[\uDD0C\uDD0F\uDD18-\uDD1F\uDD26\uDD30-\uDD39\uDD3C-\uDD3E\uDD77\uDDB5\uDDB6\uDDB8\uDDB9\uDDBB\uDDCD-\uDDCF\uDDD1-\uDDDD])/g;
};
const FD = O(uD);
function A$1(t, u$1 = {}) {
	if (typeof t != "string" || t.length === 0 || (u$1 = {
		ambiguousIsNarrow: !0,
		...u$1
	}, t = T$1(t), t.length === 0)) return 0;
	t = t.replace(FD(), "  ");
	const F$1 = u$1.ambiguousIsNarrow ? 1 : 2;
	let e = 0;
	for (const s of t) {
		const i = s.codePointAt(0);
		if (i <= 31 || i >= 127 && i <= 159 || i >= 768 && i <= 879) continue;
		switch (DD.eastAsianWidth(s)) {
			case "F":
			case "W":
				e += 2;
				break;
			case "A":
				e += F$1;
				break;
			default: e += 1;
		}
	}
	return e;
}
const m = 10, L$1 = (t = 0) => (u$1) => `\x1B[${u$1 + t}m`, N = (t = 0) => (u$1) => `\x1B[${38 + t};5;${u$1}m`, I = (t = 0) => (u$1, F$1, e) => `\x1B[${38 + t};2;${u$1};${F$1};${e}m`, r = {
	modifier: {
		reset: [0, 0],
		bold: [1, 22],
		dim: [2, 22],
		italic: [3, 23],
		underline: [4, 24],
		overline: [53, 55],
		inverse: [7, 27],
		hidden: [8, 28],
		strikethrough: [9, 29]
	},
	color: {
		black: [30, 39],
		red: [31, 39],
		green: [32, 39],
		yellow: [33, 39],
		blue: [34, 39],
		magenta: [35, 39],
		cyan: [36, 39],
		white: [37, 39],
		blackBright: [90, 39],
		gray: [90, 39],
		grey: [90, 39],
		redBright: [91, 39],
		greenBright: [92, 39],
		yellowBright: [93, 39],
		blueBright: [94, 39],
		magentaBright: [95, 39],
		cyanBright: [96, 39],
		whiteBright: [97, 39]
	},
	bgColor: {
		bgBlack: [40, 49],
		bgRed: [41, 49],
		bgGreen: [42, 49],
		bgYellow: [43, 49],
		bgBlue: [44, 49],
		bgMagenta: [45, 49],
		bgCyan: [46, 49],
		bgWhite: [47, 49],
		bgBlackBright: [100, 49],
		bgGray: [100, 49],
		bgGrey: [100, 49],
		bgRedBright: [101, 49],
		bgGreenBright: [102, 49],
		bgYellowBright: [103, 49],
		bgBlueBright: [104, 49],
		bgMagentaBright: [105, 49],
		bgCyanBright: [106, 49],
		bgWhiteBright: [107, 49]
	}
};
Object.keys(r.modifier);
const tD = Object.keys(r.color), eD = Object.keys(r.bgColor);
[...tD, ...eD];
function sD() {
	const t = new Map();
	for (const [u$1, F$1] of Object.entries(r)) {
		for (const [e, s] of Object.entries(F$1)) r[e] = {
			open: `\x1B[${s[0]}m`,
			close: `\x1B[${s[1]}m`
		}, F$1[e] = r[e], t.set(s[0], s[1]);
		Object.defineProperty(r, u$1, {
			value: F$1,
			enumerable: !1
		});
	}
	return Object.defineProperty(r, "codes", {
		value: t,
		enumerable: !1
	}), r.color.close = "\x1B[39m", r.bgColor.close = "\x1B[49m", r.color.ansi = L$1(), r.color.ansi256 = N(), r.color.ansi16m = I(), r.bgColor.ansi = L$1(m), r.bgColor.ansi256 = N(m), r.bgColor.ansi16m = I(m), Object.defineProperties(r, {
		rgbToAnsi256: {
			value: (u$1, F$1, e) => u$1 === F$1 && F$1 === e ? u$1 < 8 ? 16 : u$1 > 248 ? 231 : Math.round((u$1 - 8) / 247 * 24) + 232 : 16 + 36 * Math.round(u$1 / 255 * 5) + 6 * Math.round(F$1 / 255 * 5) + Math.round(e / 255 * 5),
			enumerable: !1
		},
		hexToRgb: {
			value: (u$1) => {
				const F$1 = /[a-f\d]{6}|[a-f\d]{3}/i.exec(u$1.toString(16));
				if (!F$1) return [
					0,
					0,
					0
				];
				let [e] = F$1;
				e.length === 3 && (e = [...e].map((i) => i + i).join(""));
				const s = Number.parseInt(e, 16);
				return [
					s >> 16 & 255,
					s >> 8 & 255,
					s & 255
				];
			},
			enumerable: !1
		},
		hexToAnsi256: {
			value: (u$1) => r.rgbToAnsi256(...r.hexToRgb(u$1)),
			enumerable: !1
		},
		ansi256ToAnsi: {
			value: (u$1) => {
				if (u$1 < 8) return 30 + u$1;
				if (u$1 < 16) return 90 + (u$1 - 8);
				let F$1, e, s;
				if (u$1 >= 232) F$1 = ((u$1 - 232) * 10 + 8) / 255, e = F$1, s = F$1;
				else {
					u$1 -= 16;
					const C$1 = u$1 % 36;
					F$1 = Math.floor(u$1 / 36) / 5, e = Math.floor(C$1 / 6) / 5, s = C$1 % 6 / 5;
				}
				const i = Math.max(F$1, e, s) * 2;
				if (i === 0) return 30;
				let D$1 = 30 + (Math.round(s) << 2 | Math.round(e) << 1 | Math.round(F$1));
				return i === 2 && (D$1 += 60), D$1;
			},
			enumerable: !1
		},
		rgbToAnsi: {
			value: (u$1, F$1, e) => r.ansi256ToAnsi(r.rgbToAnsi256(u$1, F$1, e)),
			enumerable: !1
		},
		hexToAnsi: {
			value: (u$1) => r.ansi256ToAnsi(r.hexToAnsi256(u$1)),
			enumerable: !1
		}
	}), r;
}
const iD = sD(), v = new Set(["\x1B", ""]), CD = 39, w$1 = "\x07", W$1 = "[", rD = "]", R = "m", y = `${rD}8;;`, V$1 = (t) => `${v.values().next().value}${W$1}${t}${R}`, z = (t) => `${v.values().next().value}${y}${t}${w$1}`, ED = (t) => t.split(" ").map((u$1) => A$1(u$1)), _$1 = (t, u$1, F$1) => {
	const e = [...u$1];
	let s = !1, i = !1, D$1 = A$1(T$1(t[t.length - 1]));
	for (const [C$1, o$1] of e.entries()) {
		const E = A$1(o$1);
		if (D$1 + E <= F$1 ? t[t.length - 1] += o$1 : (t.push(o$1), D$1 = 0), v.has(o$1) && (s = !0, i = e.slice(C$1 + 1).join("").startsWith(y)), s) {
			i ? o$1 === w$1 && (s = !1, i = !1) : o$1 === R && (s = !1);
			continue;
		}
		D$1 += E, D$1 === F$1 && C$1 < e.length - 1 && (t.push(""), D$1 = 0);
	}
	!D$1 && t[t.length - 1].length > 0 && t.length > 1 && (t[t.length - 2] += t.pop());
}, nD = (t) => {
	const u$1 = t.split(" ");
	let F$1 = u$1.length;
	for (; F$1 > 0 && !(A$1(u$1[F$1 - 1]) > 0);) F$1--;
	return F$1 === u$1.length ? t : u$1.slice(0, F$1).join(" ") + u$1.slice(F$1).join("");
}, oD = (t, u$1, F$1 = {}) => {
	if (F$1.trim !== !1 && t.trim() === "") return "";
	let e = "", s, i;
	const D$1 = ED(t);
	let C$1 = [""];
	for (const [E, a] of t.split(" ").entries()) {
		F$1.trim !== !1 && (C$1[C$1.length - 1] = C$1[C$1.length - 1].trimStart());
		let n = A$1(C$1[C$1.length - 1]);
		if (E !== 0 && (n >= u$1 && (F$1.wordWrap === !1 || F$1.trim === !1) && (C$1.push(""), n = 0), (n > 0 || F$1.trim === !1) && (C$1[C$1.length - 1] += " ", n++)), F$1.hard && D$1[E] > u$1) {
			const B$1 = u$1 - n, p$1 = 1 + Math.floor((D$1[E] - B$1 - 1) / u$1);
			Math.floor((D$1[E] - 1) / u$1) < p$1 && C$1.push(""), _$1(C$1, a, u$1);
			continue;
		}
		if (n + D$1[E] > u$1 && n > 0 && D$1[E] > 0) {
			if (F$1.wordWrap === !1 && n < u$1) {
				_$1(C$1, a, u$1);
				continue;
			}
			C$1.push("");
		}
		if (n + D$1[E] > u$1 && F$1.wordWrap === !1) {
			_$1(C$1, a, u$1);
			continue;
		}
		C$1[C$1.length - 1] += a;
	}
	F$1.trim !== !1 && (C$1 = C$1.map((E) => nD(E)));
	const o$1 = [...C$1.join(`
`)];
	for (const [E, a] of o$1.entries()) {
		if (e += a, v.has(a)) {
			const { groups: B$1 } = new RegExp(`(?:\\${W$1}(?<code>\\d+)m|\\${y}(?<uri>.*)${w$1})`).exec(o$1.slice(E).join("")) || { groups: {} };
			if (B$1.code !== void 0) {
				const p$1 = Number.parseFloat(B$1.code);
				s = p$1 === CD ? void 0 : p$1;
			} else B$1.uri !== void 0 && (i = B$1.uri.length === 0 ? void 0 : B$1.uri);
		}
		const n = iD.codes.get(Number(s));
		o$1[E + 1] === `
` ? (i && (e += z("")), s && n && (e += V$1(n))) : a === `
` && (s && n && (e += V$1(s)), i && (e += z(i)));
	}
	return e;
};
function G(t, u$1, F$1) {
	return String(t).normalize().replace(/\r\n/g, `
`).split(`
`).map((e) => oD(e, u$1, F$1)).join(`
`);
}
const aD = [
	"up",
	"down",
	"left",
	"right",
	"space",
	"enter",
	"cancel"
], c = {
	actions: new Set(aD),
	aliases: new Map([
		["k", "up"],
		["j", "down"],
		["h", "left"],
		["l", "right"],
		["", "cancel"],
		["escape", "cancel"]
	])
};
function k$1(t, u$1) {
	if (typeof t == "string") return c.aliases.get(t) === u$1;
	for (const F$1 of t) if (F$1 !== void 0 && k$1(F$1, u$1)) return !0;
	return !1;
}
function lD(t, u$1) {
	if (t === u$1) return;
	const F$1 = t.split(`
`), e = u$1.split(`
`), s = [];
	for (let i = 0; i < Math.max(F$1.length, e.length); i++) F$1[i] !== e[i] && s.push(i);
	return s;
}
const xD = globalThis.process.platform.startsWith("win"), S = Symbol("clack:cancel");
function BD(t) {
	return t === S;
}
function d$1(t, u$1) {
	const F$1 = t;
	F$1.isTTY && F$1.setRawMode(u$1);
}
function cD({ input: t = stdin, output: u$1 = stdout, overwrite: F$1 = !0, hideCursor: e = !0 } = {}) {
	const s = f$1.createInterface({
		input: t,
		output: u$1,
		prompt: "",
		tabSize: 1
	});
	f$1.emitKeypressEvents(t, s), t.isTTY && t.setRawMode(!0);
	const i = (D$1, { name: C$1, sequence: o$1 }) => {
		const E = String(D$1);
		if (k$1([
			E,
			C$1,
			o$1
		], "cancel")) {
			e && u$1.write(import_src.cursor.show), process.exit(0);
			return;
		}
		if (!F$1) return;
		const a = C$1 === "return" ? 0 : -1, n = C$1 === "return" ? -1 : 0;
		f$1.moveCursor(u$1, a, n, () => {
			f$1.clearLine(u$1, 1, () => {
				t.once("keypress", i);
			});
		});
	};
	return e && u$1.write(import_src.cursor.hide), t.once("keypress", i), () => {
		t.off("keypress", i), e && u$1.write(import_src.cursor.show), t.isTTY && !xD && t.setRawMode(!1), s.terminal = !1, s.close();
	};
}
var AD = Object.defineProperty, pD = (t, u$1, F$1) => u$1 in t ? AD(t, u$1, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: F$1
}) : t[u$1] = F$1, h = (t, u$1, F$1) => (pD(t, typeof u$1 != "symbol" ? u$1 + "" : u$1, F$1), F$1);
var x = class {
	constructor(u$1, F$1 = !0) {
		h(this, "input"), h(this, "output"), h(this, "_abortSignal"), h(this, "rl"), h(this, "opts"), h(this, "_render"), h(this, "_track", !1), h(this, "_prevFrame", ""), h(this, "_subscribers", new Map()), h(this, "_cursor", 0), h(this, "state", "initial"), h(this, "error", ""), h(this, "value");
		const { input: e = stdin, output: s = stdout, render: i, signal: D$1,...C$1 } = u$1;
		this.opts = C$1, this.onKeypress = this.onKeypress.bind(this), this.close = this.close.bind(this), this.render = this.render.bind(this), this._render = i.bind(this), this._track = F$1, this._abortSignal = D$1, this.input = e, this.output = s;
	}
	unsubscribe() {
		this._subscribers.clear();
	}
	setSubscriber(u$1, F$1) {
		const e = this._subscribers.get(u$1) ?? [];
		e.push(F$1), this._subscribers.set(u$1, e);
	}
	on(u$1, F$1) {
		this.setSubscriber(u$1, { cb: F$1 });
	}
	once(u$1, F$1) {
		this.setSubscriber(u$1, {
			cb: F$1,
			once: !0
		});
	}
	emit(u$1, ...F$1) {
		const e = this._subscribers.get(u$1) ?? [], s = [];
		for (const i of e) i.cb(...F$1), i.once && s.push(() => e.splice(e.indexOf(i), 1));
		for (const i of s) i();
	}
	prompt() {
		return new Promise((u$1, F$1) => {
			if (this._abortSignal) {
				if (this._abortSignal.aborted) return this.state = "cancel", this.close(), u$1(S);
				this._abortSignal.addEventListener("abort", () => {
					this.state = "cancel", this.close();
				}, { once: !0 });
			}
			const e = new WriteStream(0);
			e._write = (s, i, D$1) => {
				this._track && (this.value = this.rl?.line.replace(/\t/g, ""), this._cursor = this.rl?.cursor ?? 0, this.emit("value", this.value)), D$1();
			}, this.input.pipe(e), this.rl = M.createInterface({
				input: this.input,
				output: e,
				tabSize: 2,
				prompt: "",
				escapeCodeTimeout: 50
			}), M.emitKeypressEvents(this.input, this.rl), this.rl.prompt(), this.opts.initialValue !== void 0 && this._track && this.rl.write(this.opts.initialValue), this.input.on("keypress", this.onKeypress), d$1(this.input, !0), this.output.on("resize", this.render), this.render(), this.once("submit", () => {
				this.output.write(import_src.cursor.show), this.output.off("resize", this.render), d$1(this.input, !1), u$1(this.value);
			}), this.once("cancel", () => {
				this.output.write(import_src.cursor.show), this.output.off("resize", this.render), d$1(this.input, !1), u$1(S);
			});
		});
	}
	onKeypress(u$1, F$1) {
		if (this.state === "error" && (this.state = "active"), F$1?.name && (!this._track && c.aliases.has(F$1.name) && this.emit("cursor", c.aliases.get(F$1.name)), c.actions.has(F$1.name) && this.emit("cursor", F$1.name)), u$1 && (u$1.toLowerCase() === "y" || u$1.toLowerCase() === "n") && this.emit("confirm", u$1.toLowerCase() === "y"), u$1 === "	" && this.opts.placeholder && (this.value || (this.rl?.write(this.opts.placeholder), this.emit("value", this.opts.placeholder))), u$1 && this.emit("key", u$1.toLowerCase()), F$1?.name === "return") {
			if (this.opts.validate) {
				const e = this.opts.validate(this.value);
				e && (this.error = e instanceof Error ? e.message : e, this.state = "error", this.rl?.write(this.value));
			}
			this.state !== "error" && (this.state = "submit");
		}
		k$1([
			u$1,
			F$1?.name,
			F$1?.sequence
		], "cancel") && (this.state = "cancel"), (this.state === "submit" || this.state === "cancel") && this.emit("finalize"), this.render(), (this.state === "submit" || this.state === "cancel") && this.close();
	}
	close() {
		this.input.unpipe(), this.input.removeListener("keypress", this.onKeypress), this.output.write(`
`), d$1(this.input, !1), this.rl?.close(), this.rl = void 0, this.emit(`${this.state}`, this.value), this.unsubscribe();
	}
	restoreCursor() {
		const u$1 = G(this._prevFrame, process.stdout.columns, { hard: !0 }).split(`
`).length - 1;
		this.output.write(import_src.cursor.move(-999, u$1 * -1));
	}
	render() {
		const u$1 = G(this._render(this) ?? "", process.stdout.columns, { hard: !0 });
		if (u$1 !== this._prevFrame) {
			if (this.state === "initial") this.output.write(import_src.cursor.hide);
			else {
				const F$1 = lD(this._prevFrame, u$1);
				if (this.restoreCursor(), F$1 && F$1?.length === 1) {
					const e = F$1[0];
					this.output.write(import_src.cursor.move(0, e)), this.output.write(import_src.erase.lines(1));
					const s = u$1.split(`
`);
					this.output.write(s[e]), this._prevFrame = u$1, this.output.write(import_src.cursor.move(0, s.length - e - 1));
					return;
				}
				if (F$1 && F$1?.length > 1) {
					const e = F$1[0];
					this.output.write(import_src.cursor.move(0, e)), this.output.write(import_src.erase.down());
					const s = u$1.split(`
`).slice(e);
					this.output.write(s.join(`
`)), this._prevFrame = u$1;
					return;
				}
				this.output.write(import_src.erase.down());
			}
			this.output.write(u$1), this.state === "initial" && (this.state = "active"), this._prevFrame = u$1;
		}
	}
};
var fD = class extends x {
	get cursor() {
		return this.value ? 0 : 1;
	}
	get _value() {
		return this.cursor === 0;
	}
	constructor(u$1) {
		super(u$1, !1), this.value = !!u$1.initialValue, this.on("value", () => {
			this.value = this._value;
		}), this.on("confirm", (F$1) => {
			this.output.write(import_src.cursor.move(0, -1)), this.value = F$1, this.state = "submit", this.close();
		}), this.on("cursor", () => {
			this.value = !this.value;
		});
	}
};
var gD = Object.defineProperty, vD = (t, u$1, F$1) => u$1 in t ? gD(t, u$1, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: F$1
}) : t[u$1] = F$1, K$1 = (t, u$1, F$1) => (vD(t, typeof u$1 != "symbol" ? u$1 + "" : u$1, F$1), F$1);
let dD = class extends x {
	constructor(u$1) {
		super(u$1, !1), K$1(this, "options"), K$1(this, "cursor", 0);
		const { options: F$1 } = u$1;
		this.options = Object.entries(F$1).flatMap(([e, s]) => [{
			value: e,
			group: !0,
			label: e
		}, ...s.map((i) => ({
			...i,
			group: e
		}))]), this.value = [...u$1.initialValues ?? []], this.cursor = Math.max(this.options.findIndex(({ value: e }) => e === u$1.cursorAt), 0), this.on("cursor", (e) => {
			switch (e) {
				case "left":
				case "up":
					this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1;
					break;
				case "down":
				case "right":
					this.cursor = this.cursor === this.options.length - 1 ? 0 : this.cursor + 1;
					break;
				case "space":
					this.toggleValue();
					break;
			}
		});
	}
	getGroupItems(u$1) {
		return this.options.filter((F$1) => F$1.group === u$1);
	}
	isGroupSelected(u$1) {
		return this.getGroupItems(u$1).every((F$1) => this.value.includes(F$1.value));
	}
	toggleValue() {
		const u$1 = this.options[this.cursor];
		if (u$1.group === !0) {
			const F$1 = u$1.value, e = this.getGroupItems(F$1);
			this.isGroupSelected(F$1) ? this.value = this.value.filter((s) => e.findIndex((i) => i.value === s) === -1) : this.value = [...this.value, ...e.map((s) => s.value)], this.value = Array.from(new Set(this.value));
		} else {
			const F$1 = this.value.includes(u$1.value);
			this.value = F$1 ? this.value.filter((e) => e !== u$1.value) : [...this.value, u$1.value];
		}
	}
};
var bD = Object.defineProperty, mD = (t, u$1, F$1) => u$1 in t ? bD(t, u$1, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: F$1
}) : t[u$1] = F$1, Y$1 = (t, u$1, F$1) => (mD(t, typeof u$1 != "symbol" ? u$1 + "" : u$1, F$1), F$1);
let wD = class extends x {
	constructor(u$1) {
		super(u$1, !1), Y$1(this, "options"), Y$1(this, "cursor", 0), this.options = u$1.options, this.value = [...u$1.initialValues ?? []], this.cursor = Math.max(this.options.findIndex(({ value: F$1 }) => F$1 === u$1.cursorAt), 0), this.on("key", (F$1) => {
			F$1 === "a" && this.toggleAll();
		}), this.on("cursor", (F$1) => {
			switch (F$1) {
				case "left":
				case "up":
					this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1;
					break;
				case "down":
				case "right":
					this.cursor = this.cursor === this.options.length - 1 ? 0 : this.cursor + 1;
					break;
				case "space":
					this.toggleValue();
					break;
			}
		});
	}
	get _value() {
		return this.options[this.cursor].value;
	}
	toggleAll() {
		const u$1 = this.value.length === this.options.length;
		this.value = u$1 ? [] : this.options.map((F$1) => F$1.value);
	}
	toggleValue() {
		const u$1 = this.value.includes(this._value);
		this.value = u$1 ? this.value.filter((F$1) => F$1 !== this._value) : [...this.value, this._value];
	}
};
var yD = Object.defineProperty, _D = (t, u$1, F$1) => u$1 in t ? yD(t, u$1, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: F$1
}) : t[u$1] = F$1, Z = (t, u$1, F$1) => (_D(t, typeof u$1 != "symbol" ? u$1 + "" : u$1, F$1), F$1);
var kD = class extends x {
	constructor({ mask: u$1,...F$1 }) {
		super(F$1), Z(this, "valueWithCursor", ""), Z(this, "_mask", "•"), this._mask = u$1 ?? "•", this.on("finalize", () => {
			this.valueWithCursor = this.masked;
		}), this.on("value", () => {
			if (this.cursor >= this.value.length) this.valueWithCursor = `${this.masked}${import_picocolors.default.inverse(import_picocolors.default.hidden("_"))}`;
			else {
				const e = this.masked.slice(0, this.cursor), s = this.masked.slice(this.cursor);
				this.valueWithCursor = `${e}${import_picocolors.default.inverse(s[0])}${s.slice(1)}`;
			}
		});
	}
	get cursor() {
		return this._cursor;
	}
	get masked() {
		return this.value.replaceAll(/./g, this._mask);
	}
};
var SD = Object.defineProperty, $D = (t, u$1, F$1) => u$1 in t ? SD(t, u$1, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: F$1
}) : t[u$1] = F$1, q$1 = (t, u$1, F$1) => ($D(t, typeof u$1 != "symbol" ? u$1 + "" : u$1, F$1), F$1);
var jD = class extends x {
	constructor(u$1) {
		super(u$1, !1), q$1(this, "options"), q$1(this, "cursor", 0), this.options = u$1.options, this.cursor = this.options.findIndex(({ value: F$1 }) => F$1 === u$1.initialValue), this.cursor === -1 && (this.cursor = 0), this.changeValue(), this.on("cursor", (F$1) => {
			switch (F$1) {
				case "left":
				case "up":
					this.cursor = this.cursor === 0 ? this.options.length - 1 : this.cursor - 1;
					break;
				case "down":
				case "right":
					this.cursor = this.cursor === this.options.length - 1 ? 0 : this.cursor + 1;
					break;
			}
			this.changeValue();
		});
	}
	get _value() {
		return this.options[this.cursor];
	}
	changeValue() {
		this.value = this._value.value;
	}
};
var MD = Object.defineProperty, TD = (t, u$1, F$1) => u$1 in t ? MD(t, u$1, {
	enumerable: !0,
	configurable: !0,
	writable: !0,
	value: F$1
}) : t[u$1] = F$1, H = (t, u$1, F$1) => (TD(t, typeof u$1 != "symbol" ? u$1 + "" : u$1, F$1), F$1);
var OD = class extends x {
	constructor(u$1) {
		super(u$1, !1), H(this, "options"), H(this, "cursor", 0), this.options = u$1.options;
		const F$1 = this.options.map(({ value: [e] }) => e?.toLowerCase());
		this.cursor = Math.max(F$1.indexOf(u$1.initialValue), 0), this.on("key", (e) => {
			if (!F$1.includes(e)) return;
			const s = this.options.find(({ value: [i] }) => i?.toLowerCase() === e);
			s && (this.value = s.value, this.state = "submit", this.emit("submit"));
		});
	}
};
var PD = class extends x {
	get valueWithCursor() {
		if (this.state === "submit") return this.value;
		if (this.cursor >= this.value.length) return `${this.value}\u2588`;
		const u$1 = this.value.slice(0, this.cursor), [F$1, ...e] = this.value.slice(this.cursor);
		return `${u$1}${import_picocolors.default.inverse(F$1)}${e.join("")}`;
	}
	get cursor() {
		return this._cursor;
	}
	constructor(u$1) {
		super(u$1), this.on("finalize", () => {
			this.value || (this.value = u$1.defaultValue);
		});
	}
};

//#endregion
//#region node_modules/.pnpm/@clack+prompts@0.10.0/node_modules/@clack/prompts/dist/index.mjs
function ce() {
	return g.platform !== "win32" ? g.env.TERM !== "linux" : !!g.env.CI || !!g.env.WT_SESSION || !!g.env.TERMINUS_SUBLIME || g.env.ConEmuTask === "{cmd::Cmder}" || g.env.TERM_PROGRAM === "Terminus-Sublime" || g.env.TERM_PROGRAM === "vscode" || g.env.TERM === "xterm-256color" || g.env.TERM === "alacritty" || g.env.TERMINAL_EMULATOR === "JetBrains-JediTerm";
}
const V = ce(), u = (t, n) => V ? t : n, le = u("◆", "*"), L = u("■", "x"), W = u("▲", "x"), C = u("◇", "o"), ue = u("┌", "T"), o = u("│", "|"), d = u("└", "—"), k = u("●", ">"), P = u("○", " "), A = u("◻", "[•]"), T = u("◼", "[+]"), F = u("◻", "[ ]"), $e = u("▪", "•"), _ = u("─", "-"), me = u("╮", "+"), de = u("├", "+"), pe = u("╯", "+"), q = u("●", "•"), D = u("◆", "*"), U = u("▲", "!"), K = u("■", "x"), w = (t) => {
	switch (t) {
		case "initial":
		case "active": return import_picocolors.default.cyan(le);
		case "cancel": return import_picocolors.default.red(L);
		case "error": return import_picocolors.default.yellow(W);
		case "submit": return import_picocolors.default.green(C);
	}
}, B = (t) => {
	const { cursor: n, options: s, style: r$1 } = t, i = t.maxItems ?? Number.POSITIVE_INFINITY, a = Math.max(process.stdout.rows - 4, 0), c$1 = Math.min(a, Math.max(i, 5));
	let l = 0;
	n >= l + c$1 - 3 ? l = Math.max(Math.min(n - c$1 + 3, s.length - c$1), 0) : n < l + 2 && (l = Math.max(n - 2, 0));
	const $ = c$1 < s.length && l > 0, p$1 = c$1 < s.length && l + c$1 < s.length;
	return s.slice(l, l + c$1).map((M$1, v$1, x$1) => {
		const j = v$1 === 0 && $, E = v$1 === x$1.length - 1 && p$1;
		return j || E ? import_picocolors.default.dim("...") : r$1(M$1, v$1 + l === n);
	});
}, he = (t) => new PD({
	validate: t.validate,
	placeholder: t.placeholder,
	defaultValue: t.defaultValue,
	initialValue: t.initialValue,
	render() {
		const n = `${import_picocolors.default.gray(o)}
${w(this.state)}  ${t.message}
`, s = t.placeholder ? import_picocolors.default.inverse(t.placeholder[0]) + import_picocolors.default.dim(t.placeholder.slice(1)) : import_picocolors.default.inverse(import_picocolors.default.hidden("_")), r$1 = this.value ? this.valueWithCursor : s;
		switch (this.state) {
			case "error": return `${n.trim()}
${import_picocolors.default.yellow(o)}  ${r$1}
${import_picocolors.default.yellow(d)}  ${import_picocolors.default.yellow(this.error)}
`;
			case "submit": return `${n}${import_picocolors.default.gray(o)}  ${import_picocolors.default.dim(this.value || t.placeholder)}`;
			case "cancel": return `${n}${import_picocolors.default.gray(o)}  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(this.value ?? ""))}${this.value?.trim() ? `
${import_picocolors.default.gray(o)}` : ""}`;
			default: return `${n}${import_picocolors.default.cyan(o)}  ${r$1}
${import_picocolors.default.cyan(d)}
`;
		}
	}
}).prompt(), ge = (t) => new kD({
	validate: t.validate,
	mask: t.mask ?? $e,
	render() {
		const n = `${import_picocolors.default.gray(o)}
${w(this.state)}  ${t.message}
`, s = this.valueWithCursor, r$1 = this.masked;
		switch (this.state) {
			case "error": return `${n.trim()}
${import_picocolors.default.yellow(o)}  ${r$1}
${import_picocolors.default.yellow(d)}  ${import_picocolors.default.yellow(this.error)}
`;
			case "submit": return `${n}${import_picocolors.default.gray(o)}  ${import_picocolors.default.dim(r$1)}`;
			case "cancel": return `${n}${import_picocolors.default.gray(o)}  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(r$1 ?? ""))}${r$1 ? `
${import_picocolors.default.gray(o)}` : ""}`;
			default: return `${n}${import_picocolors.default.cyan(o)}  ${s}
${import_picocolors.default.cyan(d)}
`;
		}
	}
}).prompt(), ye = (t) => {
	const n = t.active ?? "Yes", s = t.inactive ?? "No";
	return new fD({
		active: n,
		inactive: s,
		initialValue: t.initialValue ?? !0,
		render() {
			const r$1 = `${import_picocolors.default.gray(o)}
${w(this.state)}  ${t.message}
`, i = this.value ? n : s;
			switch (this.state) {
				case "submit": return `${r$1}${import_picocolors.default.gray(o)}  ${import_picocolors.default.dim(i)}`;
				case "cancel": return `${r$1}${import_picocolors.default.gray(o)}  ${import_picocolors.default.strikethrough(import_picocolors.default.dim(i))}
${import_picocolors.default.gray(o)}`;
				default: return `${r$1}${import_picocolors.default.cyan(o)}  ${this.value ? `${import_picocolors.default.green(k)} ${n}` : `${import_picocolors.default.dim(P)} ${import_picocolors.default.dim(n)}`} ${import_picocolors.default.dim("/")} ${this.value ? `${import_picocolors.default.dim(P)} ${import_picocolors.default.dim(s)}` : `${import_picocolors.default.green(k)} ${s}`}
${import_picocolors.default.cyan(d)}
`;
			}
		}
	}).prompt();
}, ve = (t) => {
	const n = (s, r$1) => {
		const i = s.label ?? String(s.value);
		switch (r$1) {
			case "selected": return `${import_picocolors.default.dim(i)}`;
			case "active": return `${import_picocolors.default.green(k)} ${i} ${s.hint ? import_picocolors.default.dim(`(${s.hint})`) : ""}`;
			case "cancelled": return `${import_picocolors.default.strikethrough(import_picocolors.default.dim(i))}`;
			default: return `${import_picocolors.default.dim(P)} ${import_picocolors.default.dim(i)}`;
		}
	};
	return new jD({
		options: t.options,
		initialValue: t.initialValue,
		render() {
			const s = `${import_picocolors.default.gray(o)}
${w(this.state)}  ${t.message}
`;
			switch (this.state) {
				case "submit": return `${s}${import_picocolors.default.gray(o)}  ${n(this.options[this.cursor], "selected")}`;
				case "cancel": return `${s}${import_picocolors.default.gray(o)}  ${n(this.options[this.cursor], "cancelled")}
${import_picocolors.default.gray(o)}`;
				default: return `${s}${import_picocolors.default.cyan(o)}  ${B({
					cursor: this.cursor,
					options: this.options,
					maxItems: t.maxItems,
					style: (r$1, i) => n(r$1, i ? "active" : "inactive")
				}).join(`
${import_picocolors.default.cyan(o)}  `)}
${import_picocolors.default.cyan(d)}
`;
			}
		}
	}).prompt();
}, we = (t) => {
	const n = (s, r$1 = "inactive") => {
		const i = s.label ?? String(s.value);
		return r$1 === "selected" ? `${import_picocolors.default.dim(i)}` : r$1 === "cancelled" ? `${import_picocolors.default.strikethrough(import_picocolors.default.dim(i))}` : r$1 === "active" ? `${import_picocolors.default.bgCyan(import_picocolors.default.gray(` ${s.value} `))} ${i} ${s.hint ? import_picocolors.default.dim(`(${s.hint})`) : ""}` : `${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(` ${s.value} `)))} ${i} ${s.hint ? import_picocolors.default.dim(`(${s.hint})`) : ""}`;
	};
	return new OD({
		options: t.options,
		initialValue: t.initialValue,
		render() {
			const s = `${import_picocolors.default.gray(o)}
${w(this.state)}  ${t.message}
`;
			switch (this.state) {
				case "submit": return `${s}${import_picocolors.default.gray(o)}  ${n(this.options.find((r$1) => r$1.value === this.value) ?? t.options[0], "selected")}`;
				case "cancel": return `${s}${import_picocolors.default.gray(o)}  ${n(this.options[0], "cancelled")}
${import_picocolors.default.gray(o)}`;
				default: return `${s}${import_picocolors.default.cyan(o)}  ${this.options.map((r$1, i) => n(r$1, i === this.cursor ? "active" : "inactive")).join(`
${import_picocolors.default.cyan(o)}  `)}
${import_picocolors.default.cyan(d)}
`;
			}
		}
	}).prompt();
}, fe = (t) => {
	const n = (s, r$1) => {
		const i = s.label ?? String(s.value);
		return r$1 === "active" ? `${import_picocolors.default.cyan(A)} ${i} ${s.hint ? import_picocolors.default.dim(`(${s.hint})`) : ""}` : r$1 === "selected" ? `${import_picocolors.default.green(T)} ${import_picocolors.default.dim(i)}` : r$1 === "cancelled" ? `${import_picocolors.default.strikethrough(import_picocolors.default.dim(i))}` : r$1 === "active-selected" ? `${import_picocolors.default.green(T)} ${i} ${s.hint ? import_picocolors.default.dim(`(${s.hint})`) : ""}` : r$1 === "submitted" ? `${import_picocolors.default.dim(i)}` : `${import_picocolors.default.dim(F)} ${import_picocolors.default.dim(i)}`;
	};
	return new wD({
		options: t.options,
		initialValues: t.initialValues,
		required: t.required ?? !0,
		cursorAt: t.cursorAt,
		validate(s) {
			if (this.required && s.length === 0) return `Please select at least one option.
${import_picocolors.default.reset(import_picocolors.default.dim(`Press ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" space ")))} to select, ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" enter ")))} to submit`))}`;
		},
		render() {
			const s = `${import_picocolors.default.gray(o)}
${w(this.state)}  ${t.message}
`, r$1 = (i, a) => {
				const c$1 = this.value.includes(i.value);
				return a && c$1 ? n(i, "active-selected") : c$1 ? n(i, "selected") : n(i, a ? "active" : "inactive");
			};
			switch (this.state) {
				case "submit": return `${s}${import_picocolors.default.gray(o)}  ${this.options.filter(({ value: i }) => this.value.includes(i)).map((i) => n(i, "submitted")).join(import_picocolors.default.dim(", ")) || import_picocolors.default.dim("none")}`;
				case "cancel": {
					const i = this.options.filter(({ value: a }) => this.value.includes(a)).map((a) => n(a, "cancelled")).join(import_picocolors.default.dim(", "));
					return `${s}${import_picocolors.default.gray(o)}  ${i.trim() ? `${i}
${import_picocolors.default.gray(o)}` : ""}`;
				}
				case "error": {
					const i = this.error.split(`
`).map((a, c$1) => c$1 === 0 ? `${import_picocolors.default.yellow(d)}  ${import_picocolors.default.yellow(a)}` : `   ${a}`).join(`
`);
					return `${s + import_picocolors.default.yellow(o)}  ${B({
						options: this.options,
						cursor: this.cursor,
						maxItems: t.maxItems,
						style: r$1
					}).join(`
${import_picocolors.default.yellow(o)}  `)}
${i}
`;
				}
				default: return `${s}${import_picocolors.default.cyan(o)}  ${B({
					options: this.options,
					cursor: this.cursor,
					maxItems: t.maxItems,
					style: r$1
				}).join(`
${import_picocolors.default.cyan(o)}  `)}
${import_picocolors.default.cyan(d)}
`;
			}
		}
	}).prompt();
}, be = (t) => {
	const n = (s, r$1, i = []) => {
		const a = s.label ?? String(s.value), c$1 = typeof s.group == "string", l = c$1 && (i[i.indexOf(s) + 1] ?? { group: !0 }), $ = c$1 && l.group === !0, p$1 = c$1 ? `${$ ? d : o} ` : "";
		return r$1 === "active" ? `${import_picocolors.default.dim(p$1)}${import_picocolors.default.cyan(A)} ${a} ${s.hint ? import_picocolors.default.dim(`(${s.hint})`) : ""}` : r$1 === "group-active" ? `${p$1}${import_picocolors.default.cyan(A)} ${import_picocolors.default.dim(a)}` : r$1 === "group-active-selected" ? `${p$1}${import_picocolors.default.green(T)} ${import_picocolors.default.dim(a)}` : r$1 === "selected" ? `${import_picocolors.default.dim(p$1)}${import_picocolors.default.green(T)} ${import_picocolors.default.dim(a)}` : r$1 === "cancelled" ? `${import_picocolors.default.strikethrough(import_picocolors.default.dim(a))}` : r$1 === "active-selected" ? `${import_picocolors.default.dim(p$1)}${import_picocolors.default.green(T)} ${a} ${s.hint ? import_picocolors.default.dim(`(${s.hint})`) : ""}` : r$1 === "submitted" ? `${import_picocolors.default.dim(a)}` : `${import_picocolors.default.dim(p$1)}${import_picocolors.default.dim(F)} ${import_picocolors.default.dim(a)}`;
	};
	return new dD({
		options: t.options,
		initialValues: t.initialValues,
		required: t.required ?? !0,
		cursorAt: t.cursorAt,
		validate(s) {
			if (this.required && s.length === 0) return `Please select at least one option.
${import_picocolors.default.reset(import_picocolors.default.dim(`Press ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" space ")))} to select, ${import_picocolors.default.gray(import_picocolors.default.bgWhite(import_picocolors.default.inverse(" enter ")))} to submit`))}`;
		},
		render() {
			const s = `${import_picocolors.default.gray(o)}
${w(this.state)}  ${t.message}
`;
			switch (this.state) {
				case "submit": return `${s}${import_picocolors.default.gray(o)}  ${this.options.filter(({ value: r$1 }) => this.value.includes(r$1)).map((r$1) => n(r$1, "submitted")).join(import_picocolors.default.dim(", "))}`;
				case "cancel": {
					const r$1 = this.options.filter(({ value: i }) => this.value.includes(i)).map((i) => n(i, "cancelled")).join(import_picocolors.default.dim(", "));
					return `${s}${import_picocolors.default.gray(o)}  ${r$1.trim() ? `${r$1}
${import_picocolors.default.gray(o)}` : ""}`;
				}
				case "error": {
					const r$1 = this.error.split(`
`).map((i, a) => a === 0 ? `${import_picocolors.default.yellow(d)}  ${import_picocolors.default.yellow(i)}` : `   ${i}`).join(`
`);
					return `${s}${import_picocolors.default.yellow(o)}  ${this.options.map((i, a, c$1) => {
						const l = this.value.includes(i.value) || i.group === !0 && this.isGroupSelected(`${i.value}`), $ = a === this.cursor;
						return !$ && typeof i.group == "string" && this.options[this.cursor].value === i.group ? n(i, l ? "group-active-selected" : "group-active", c$1) : $ && l ? n(i, "active-selected", c$1) : l ? n(i, "selected", c$1) : n(i, $ ? "active" : "inactive", c$1);
					}).join(`
${import_picocolors.default.yellow(o)}  `)}
${r$1}
`;
				}
				default: return `${s}${import_picocolors.default.cyan(o)}  ${this.options.map((r$1, i, a) => {
					const c$1 = this.value.includes(r$1.value) || r$1.group === !0 && this.isGroupSelected(`${r$1.value}`), l = i === this.cursor;
					return !l && typeof r$1.group == "string" && this.options[this.cursor].value === r$1.group ? n(r$1, c$1 ? "group-active-selected" : "group-active", a) : l && c$1 ? n(r$1, "active-selected", a) : c$1 ? n(r$1, "selected", a) : n(r$1, l ? "active" : "inactive", a);
				}).join(`
${import_picocolors.default.cyan(o)}  `)}
${import_picocolors.default.cyan(d)}
`;
			}
		}
	}).prompt();
}, Me = (t = "", n = "") => {
	const s = `
${t}
`.split(`
`), r$1 = stripVTControlCharacters(n).length, i = Math.max(s.reduce((c$1, l) => {
		const $ = stripVTControlCharacters(l);
		return $.length > c$1 ? $.length : c$1;
	}, 0), r$1) + 2, a = s.map((c$1) => `${import_picocolors.default.gray(o)}  ${import_picocolors.default.dim(c$1)}${" ".repeat(i - stripVTControlCharacters(c$1).length)}${import_picocolors.default.gray(o)}`).join(`
`);
	process.stdout.write(`${import_picocolors.default.gray(o)}
${import_picocolors.default.green(C)}  ${import_picocolors.default.reset(n)} ${import_picocolors.default.gray(_.repeat(Math.max(i - r$1 - 1, 1)) + me)}
${a}
${import_picocolors.default.gray(de + _.repeat(i + 2) + pe)}
`);
}, xe = (t = "") => {
	process.stdout.write(`${import_picocolors.default.gray(d)}  ${import_picocolors.default.red(t)}

`);
}, Ie = (t = "") => {
	process.stdout.write(`${import_picocolors.default.gray(ue)}  ${t}
`);
}, Se = (t = "") => {
	process.stdout.write(`${import_picocolors.default.gray(o)}
${import_picocolors.default.gray(d)}  ${t}

`);
}, f = {
	message: (t = "", { symbol: n = import_picocolors.default.gray(o) } = {}) => {
		const s = [`${import_picocolors.default.gray(o)}`];
		if (t) {
			const [r$1, ...i] = t.split(`
`);
			s.push(`${n}  ${r$1}`, ...i.map((a) => `${import_picocolors.default.gray(o)}  ${a}`));
		}
		process.stdout.write(`${s.join(`
`)}
`);
	},
	info: (t) => {
		f.message(t, { symbol: import_picocolors.default.blue(q) });
	},
	success: (t) => {
		f.message(t, { symbol: import_picocolors.default.green(D) });
	},
	step: (t) => {
		f.message(t, { symbol: import_picocolors.default.green(C) });
	},
	warn: (t) => {
		f.message(t, { symbol: import_picocolors.default.yellow(U) });
	},
	warning: (t) => {
		f.warn(t);
	},
	error: (t) => {
		f.message(t, { symbol: import_picocolors.default.red(K) });
	}
}, J = `${import_picocolors.default.gray(o)}  `, b = {
	message: async (t, { symbol: n = import_picocolors.default.gray(o) } = {}) => {
		process.stdout.write(`${import_picocolors.default.gray(o)}
${n}  `);
		let s = 3;
		for await (let r$1 of t) {
			r$1 = r$1.replace(/\n/g, `
${J}`), r$1.includes(`
`) && (s = 3 + stripVTControlCharacters(r$1.slice(r$1.lastIndexOf(`
`))).length);
			const i = stripVTControlCharacters(r$1).length;
			s + i < process.stdout.columns ? (s += i, process.stdout.write(r$1)) : (process.stdout.write(`
${J}${r$1.trimStart()}`), s = 3 + stripVTControlCharacters(r$1.trimStart()).length);
		}
		process.stdout.write(`
`);
	},
	info: (t) => b.message(t, { symbol: import_picocolors.default.blue(q) }),
	success: (t) => b.message(t, { symbol: import_picocolors.default.green(D) }),
	step: (t) => b.message(t, { symbol: import_picocolors.default.green(C) }),
	warn: (t) => b.message(t, { symbol: import_picocolors.default.yellow(U) }),
	warning: (t) => b.warn(t),
	error: (t) => b.message(t, { symbol: import_picocolors.default.red(K) })
}, Y = ({ indicator: t = "dots" } = {}) => {
	const n = V ? [
		"◒",
		"◐",
		"◓",
		"◑"
	] : [
		"•",
		"o",
		"O",
		"0"
	], s = V ? 80 : 120, r$1 = process.env.CI === "true";
	let i, a, c$1 = !1, l = "", $, p$1 = performance.now();
	const M$1 = (m$1) => {
		const h$1 = m$1 > 1 ? "Something went wrong" : "Canceled";
		c$1 && N$1(h$1, m$1);
	}, v$1 = () => M$1(2), x$1 = () => M$1(1), j = () => {
		process.on("uncaughtExceptionMonitor", v$1), process.on("unhandledRejection", v$1), process.on("SIGINT", x$1), process.on("SIGTERM", x$1), process.on("exit", M$1);
	}, E = () => {
		process.removeListener("uncaughtExceptionMonitor", v$1), process.removeListener("unhandledRejection", v$1), process.removeListener("SIGINT", x$1), process.removeListener("SIGTERM", x$1), process.removeListener("exit", M$1);
	}, O$1 = () => {
		if ($ === void 0) return;
		r$1 && process.stdout.write(`
`);
		const m$1 = $.split(`
`);
		process.stdout.write(import_src.cursor.move(-999, m$1.length - 1)), process.stdout.write(import_src.erase.down(m$1.length));
	}, R$1 = (m$1) => m$1.replace(/\.+$/, ""), G$1 = (m$1) => {
		const h$1 = (performance.now() - m$1) / 1e3, y$1 = Math.floor(h$1 / 60), I$1 = Math.floor(h$1 % 60);
		return y$1 > 0 ? `[${y$1}m ${I$1}s]` : `[${I$1}s]`;
	}, H$1 = (m$1 = "") => {
		c$1 = !0, i = cD(), l = R$1(m$1), p$1 = performance.now(), process.stdout.write(`${import_picocolors.default.gray(o)}
`);
		let h$1 = 0, y$1 = 0;
		j(), a = setInterval(() => {
			if (r$1 && l === $) return;
			O$1(), $ = l;
			const I$1 = import_picocolors.default.magenta(n[h$1]);
			if (r$1) process.stdout.write(`${I$1}  ${l}...`);
			else if (t === "timer") process.stdout.write(`${I$1}  ${l} ${G$1(p$1)}`);
			else {
				const z$1 = ".".repeat(Math.floor(y$1)).slice(0, 3);
				process.stdout.write(`${I$1}  ${l}${z$1}`);
			}
			h$1 = h$1 + 1 < n.length ? h$1 + 1 : 0, y$1 = y$1 < n.length ? y$1 + .125 : 0;
		}, s);
	}, N$1 = (m$1 = "", h$1 = 0) => {
		c$1 = !1, clearInterval(a), O$1();
		const y$1 = h$1 === 0 ? import_picocolors.default.green(C) : h$1 === 1 ? import_picocolors.default.red(L) : import_picocolors.default.red(W);
		l = R$1(m$1 ?? l), t === "timer" ? process.stdout.write(`${y$1}  ${l} ${G$1(p$1)}
`) : process.stdout.write(`${y$1}  ${l}
`), E(), i();
	};
	return {
		start: H$1,
		stop: N$1,
		message: (m$1 = "") => {
			l = R$1(m$1 ?? l);
		}
	};
}, Ce = async (t, n) => {
	const s = {}, r$1 = Object.keys(t);
	for (const i of r$1) {
		const a = t[i], c$1 = await a({ results: s })?.catch((l) => {
			throw l;
		});
		if (typeof n?.onCancel == "function" && BD(c$1)) {
			s[i] = "canceled", n.onCancel({ results: s });
			continue;
		}
		s[i] = c$1;
	}
	return s;
}, Te = async (t) => {
	for (const n of t) {
		if (n.enabled === !1) continue;
		const s = Y();
		s.start(n.title);
		const r$1 = await n.task(s.message);
		s.stop(r$1 || n.title);
	}
};

//#endregion
//#region node_modules/.pnpm/ejs@3.1.10/node_modules/ejs/lib/utils.js
var require_utils = __commonJS({ "node_modules/.pnpm/ejs@3.1.10/node_modules/ejs/lib/utils.js"(exports) {
	var regExpChars = /[|\\{}()[\]^$+*?.]/g;
	var hasOwnProperty = Object.prototype.hasOwnProperty;
	var hasOwn = function(obj, key) {
		return hasOwnProperty.apply(obj, [key]);
	};
	/**
	* Escape characters reserved in regular expressions.
	*
	* If `string` is `undefined` or `null`, the empty string is returned.
	*
	* @param {String} string Input string
	* @return {String} Escaped string
	* @static
	* @private
	*/
	exports.escapeRegExpChars = function(string) {
		if (!string) return "";
		return String(string).replace(regExpChars, "\\$&");
	};
	var _ENCODE_HTML_RULES = {
		"&": "&amp;",
		"<": "&lt;",
		">": "&gt;",
		"\"": "&#34;",
		"'": "&#39;"
	};
	var _MATCH_HTML = /[&<>'"]/g;
	function encode_char(c$1) {
		return _ENCODE_HTML_RULES[c$1] || c$1;
	}
	/**
	* Stringified version of constants used by {@link module:utils.escapeXML}.
	*
	* It is used in the process of generating {@link ClientFunction}s.
	*
	* @readonly
	* @type {String}
	*/
	var escapeFuncStr = "var _ENCODE_HTML_RULES = {\n      \"&\": \"&amp;\"\n    , \"<\": \"&lt;\"\n    , \">\": \"&gt;\"\n    , '\"': \"&#34;\"\n    , \"'\": \"&#39;\"\n    }\n  , _MATCH_HTML = /[&<>'\"]/g;\nfunction encode_char(c) {\n  return _ENCODE_HTML_RULES[c] || c;\n};\n";
	/**
	* Escape characters reserved in XML.
	*
	* If `markup` is `undefined` or `null`, the empty string is returned.
	*
	* @implements {EscapeCallback}
	* @param {String} markup Input string
	* @return {String} Escaped string
	* @static
	* @private
	*/
	exports.escapeXML = function(markup) {
		return markup == void 0 ? "" : String(markup).replace(_MATCH_HTML, encode_char);
	};
	function escapeXMLToString() {
		return Function.prototype.toString.call(this) + ";\n" + escapeFuncStr;
	}
	try {
		if (typeof Object.defineProperty === "function") Object.defineProperty(exports.escapeXML, "toString", { value: escapeXMLToString });
		else exports.escapeXML.toString = escapeXMLToString;
	} catch (err) {
		console.warn("Unable to set escapeXML.toString (is the Function prototype frozen?)");
	}
	/**
	* Naive copy of properties from one object to another.
	* Does not recurse into non-scalar properties
	* Does not check to see if the property has a value before copying
	*
	* @param  {Object} to   Destination object
	* @param  {Object} from Source object
	* @return {Object}      Destination object
	* @static
	* @private
	*/
	exports.shallowCopy = function(to, from) {
		from = from || {};
		if (to !== null && to !== void 0) for (var p$1 in from) {
			if (!hasOwn(from, p$1)) continue;
			if (p$1 === "__proto__" || p$1 === "constructor") continue;
			to[p$1] = from[p$1];
		}
		return to;
	};
	/**
	* Naive copy of a list of key names, from one object to another.
	* Only copies property if it is actually defined
	* Does not recurse into non-scalar properties
	*
	* @param  {Object} to   Destination object
	* @param  {Object} from Source object
	* @param  {Array} list List of properties to copy
	* @return {Object}      Destination object
	* @static
	* @private
	*/
	exports.shallowCopyFromList = function(to, from, list) {
		list = list || [];
		from = from || {};
		if (to !== null && to !== void 0) for (var i = 0; i < list.length; i++) {
			var p$1 = list[i];
			if (typeof from[p$1] != "undefined") {
				if (!hasOwn(from, p$1)) continue;
				if (p$1 === "__proto__" || p$1 === "constructor") continue;
				to[p$1] = from[p$1];
			}
		}
		return to;
	};
	/**
	* Simple in-process cache implementation. Does not implement limits of any
	* sort.
	*
	* @implements {Cache}
	* @static
	* @private
	*/
	exports.cache = {
		_data: {},
		set: function(key, val) {
			this._data[key] = val;
		},
		get: function(key) {
			return this._data[key];
		},
		remove: function(key) {
			delete this._data[key];
		},
		reset: function() {
			this._data = {};
		}
	};
	/**
	* Transforms hyphen case variable into camel case.
	*
	* @param {String} string Hyphen case string
	* @return {String} Camel case string
	* @static
	* @private
	*/
	exports.hyphenToCamel = function(str) {
		return str.replace(/-[a-z]/g, function(match) {
			return match[1].toUpperCase();
		});
	};
	/**
	* Returns a null-prototype object in runtimes that support it
	*
	* @return {Object} Object, prototype will be set to null where possible
	* @static
	* @private
	*/
	exports.createNullProtoObjWherePossible = function() {
		if (typeof Object.create == "function") return function() {
			return Object.create(null);
		};
		return function() {
			return {};
		};
	}();
	exports.hasOwnOnlyObject = function(obj) {
		var o$1 = exports.createNullProtoObjWherePossible();
		for (var p$1 in obj) if (hasOwn(obj, p$1)) o$1[p$1] = obj[p$1];
		return o$1;
	};
} });

//#endregion
//#region node_modules/.pnpm/ejs@3.1.10/node_modules/ejs/package.json
var require_package = __commonJS({ "node_modules/.pnpm/ejs@3.1.10/node_modules/ejs/package.json"(exports, module) {
	module.exports = {
		"name": "ejs",
		"description": "Embedded JavaScript templates",
		"keywords": [
			"template",
			"engine",
			"ejs"
		],
		"version": "3.1.10",
		"author": "Matthew Eernisse <mde@fleegix.org> (http://fleegix.org)",
		"license": "Apache-2.0",
		"bin": { "ejs": "./bin/cli.js" },
		"main": "./lib/ejs.js",
		"jsdelivr": "ejs.min.js",
		"unpkg": "ejs.min.js",
		"repository": {
			"type": "git",
			"url": "git://github.com/mde/ejs.git"
		},
		"bugs": "https://github.com/mde/ejs/issues",
		"homepage": "https://github.com/mde/ejs",
		"dependencies": { "jake": "^10.8.5" },
		"devDependencies": {
			"browserify": "^16.5.1",
			"eslint": "^6.8.0",
			"git-directory-deploy": "^1.5.1",
			"jsdoc": "^4.0.2",
			"lru-cache": "^4.0.1",
			"mocha": "^10.2.0",
			"uglify-js": "^3.3.16"
		},
		"engines": { "node": ">=0.10.0" },
		"scripts": { "test": "npx jake test" }
	};
} });

//#endregion
//#region node_modules/.pnpm/ejs@3.1.10/node_modules/ejs/lib/ejs.js
var require_ejs = __commonJS({ "node_modules/.pnpm/ejs@3.1.10/node_modules/ejs/lib/ejs.js"(exports) {
	/**
	* @file Embedded JavaScript templating engine. {@link http://ejs.co}
	* @author Matthew Eernisse <mde@fleegix.org>
	* @author Tiancheng "Timothy" Gu <timothygu99@gmail.com>
	* @project EJS
	* @license {@link http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0}
	*/
	/**
	* EJS internal functions.
	*
	* Technically this "module" lies in the same file as {@link module:ejs}, for
	* the sake of organization all the private functions re grouped into this
	* module.
	*
	* @module ejs-internal
	* @private
	*/
	/**
	* Embedded JavaScript templating engine.
	*
	* @module ejs
	* @public
	*/
	var fs$6 = __require("fs");
	var path$6 = __require("path");
	var utils = require_utils();
	var scopeOptionWarned = false;
	/** @type {string} */
	var _VERSION_STRING = require_package().version;
	var _DEFAULT_OPEN_DELIMITER = "<";
	var _DEFAULT_CLOSE_DELIMITER = ">";
	var _DEFAULT_DELIMITER = "%";
	var _DEFAULT_LOCALS_NAME = "locals";
	var _NAME = "ejs";
	var _REGEX_STRING = "(<%%|%%>|<%=|<%-|<%_|<%#|<%|%>|-%>|_%>)";
	var _OPTS_PASSABLE_WITH_DATA = [
		"delimiter",
		"scope",
		"context",
		"debug",
		"compileDebug",
		"client",
		"_with",
		"rmWhitespace",
		"strict",
		"filename",
		"async"
	];
	var _OPTS_PASSABLE_WITH_DATA_EXPRESS = _OPTS_PASSABLE_WITH_DATA.concat("cache");
	var _BOM = /^\uFEFF/;
	var _JS_IDENTIFIER = /^[a-zA-Z_$][0-9a-zA-Z_$]*$/;
	/**
	* EJS template function cache. This can be a LRU object from lru-cache NPM
	* module. By default, it is {@link module:utils.cache}, a simple in-process
	* cache that grows continuously.
	*
	* @type {Cache}
	*/
	exports.cache = utils.cache;
	/**
	* Custom file loader. Useful for template preprocessing or restricting access
	* to a certain part of the filesystem.
	*
	* @type {fileLoader}
	*/
	exports.fileLoader = fs$6.readFileSync;
	/**
	* Name of the object containing the locals.
	*
	* This variable is overridden by {@link Options}`.localsName` if it is not
	* `undefined`.
	*
	* @type {String}
	* @public
	*/
	exports.localsName = _DEFAULT_LOCALS_NAME;
	/**
	* Promise implementation -- defaults to the native implementation if available
	* This is mostly just for testability
	*
	* @type {PromiseConstructorLike}
	* @public
	*/
	exports.promiseImpl = new Function("return this;")().Promise;
	/**
	* Get the path to the included file from the parent file path and the
	* specified path.
	*
	* @param {String}  name     specified path
	* @param {String}  filename parent file path
	* @param {Boolean} [isDir=false] whether the parent file path is a directory
	* @return {String}
	*/
	exports.resolveInclude = function(name$2, filename, isDir) {
		var dirname = path$6.dirname;
		var extname = path$6.extname;
		var resolve = path$6.resolve;
		var includePath = resolve(isDir ? filename : dirname(filename), name$2);
		var ext = extname(name$2);
		if (!ext) includePath += ".ejs";
		return includePath;
	};
	/**
	* Try to resolve file path on multiple directories
	*
	* @param  {String}        name  specified path
	* @param  {Array<String>} paths list of possible parent directory paths
	* @return {String}
	*/
	function resolvePaths(name$2, paths) {
		var filePath;
		if (paths.some(function(v$1) {
			filePath = exports.resolveInclude(name$2, v$1, true);
			return fs$6.existsSync(filePath);
		})) return filePath;
	}
	/**
	* Get the path to the included file by Options
	*
	* @param  {String}  path    specified path
	* @param  {Options} options compilation options
	* @return {String}
	*/
	function getIncludePath(path$7, options) {
		var includePath;
		var filePath;
		var views = options.views;
		var match = /^[A-Za-z]+:\\|^\//.exec(path$7);
		if (match && match.length) {
			path$7 = path$7.replace(/^\/*/, "");
			if (Array.isArray(options.root)) includePath = resolvePaths(path$7, options.root);
			else includePath = exports.resolveInclude(path$7, options.root || "/", true);
		} else {
			if (options.filename) {
				filePath = exports.resolveInclude(path$7, options.filename);
				if (fs$6.existsSync(filePath)) includePath = filePath;
			}
			if (!includePath && Array.isArray(views)) includePath = resolvePaths(path$7, views);
			if (!includePath && typeof options.includer !== "function") throw new Error("Could not find the include file \"" + options.escapeFunction(path$7) + "\"");
		}
		return includePath;
	}
	/**
	* Get the template from a string or a file, either compiled on-the-fly or
	* read from cache (if enabled), and cache the template if needed.
	*
	* If `template` is not set, the file specified in `options.filename` will be
	* read.
	*
	* If `options.cache` is true, this function reads the file from
	* `options.filename` so it must be set prior to calling this function.
	*
	* @memberof module:ejs-internal
	* @param {Options} options   compilation options
	* @param {String} [template] template source
	* @return {(TemplateFunction|ClientFunction)}
	* Depending on the value of `options.client`, either type might be returned.
	* @static
	*/
	function handleCache(options, template) {
		var func;
		var filename = options.filename;
		var hasTemplate = arguments.length > 1;
		if (options.cache) {
			if (!filename) throw new Error("cache option requires a filename");
			func = exports.cache.get(filename);
			if (func) return func;
			if (!hasTemplate) template = fileLoader(filename).toString().replace(_BOM, "");
		} else if (!hasTemplate) {
			if (!filename) throw new Error("Internal EJS error: no file name or template provided");
			template = fileLoader(filename).toString().replace(_BOM, "");
		}
		func = exports.compile(template, options);
		if (options.cache) exports.cache.set(filename, func);
		return func;
	}
	/**
	* Try calling handleCache with the given options and data and call the
	* callback with the result. If an error occurs, call the callback with
	* the error. Used by renderFile().
	*
	* @memberof module:ejs-internal
	* @param {Options} options    compilation options
	* @param {Object} data        template data
	* @param {RenderFileCallback} cb callback
	* @static
	*/
	function tryHandleCache(options, data, cb) {
		var result;
		if (!cb) if (typeof exports.promiseImpl == "function") return new exports.promiseImpl(function(resolve, reject) {
			try {
				result = handleCache(options)(data);
				resolve(result);
			} catch (err) {
				reject(err);
			}
		});
		else throw new Error("Please provide a callback function");
		else {
			try {
				result = handleCache(options)(data);
			} catch (err) {
				return cb(err);
			}
			cb(null, result);
		}
	}
	/**
	* fileLoader is independent
	*
	* @param {String} filePath ejs file path.
	* @return {String} The contents of the specified file.
	* @static
	*/
	function fileLoader(filePath) {
		return exports.fileLoader(filePath);
	}
	/**
	* Get the template function.
	*
	* If `options.cache` is `true`, then the template is cached.
	*
	* @memberof module:ejs-internal
	* @param {String}  path    path for the specified file
	* @param {Options} options compilation options
	* @return {(TemplateFunction|ClientFunction)}
	* Depending on the value of `options.client`, either type might be returned
	* @static
	*/
	function includeFile(path$7, options) {
		var opts = utils.shallowCopy(utils.createNullProtoObjWherePossible(), options);
		opts.filename = getIncludePath(path$7, opts);
		if (typeof options.includer === "function") {
			var includerResult = options.includer(path$7, opts.filename);
			if (includerResult) {
				if (includerResult.filename) opts.filename = includerResult.filename;
				if (includerResult.template) return handleCache(opts, includerResult.template);
			}
		}
		return handleCache(opts);
	}
	/**
	* Re-throw the given `err` in context to the `str` of ejs, `filename`, and
	* `lineno`.
	*
	* @implements {RethrowCallback}
	* @memberof module:ejs-internal
	* @param {Error}  err      Error object
	* @param {String} str      EJS source
	* @param {String} flnm     file name of the EJS file
	* @param {Number} lineno   line number of the error
	* @param {EscapeCallback} esc
	* @static
	*/
	function rethrow(err, str, flnm, lineno, esc) {
		var lines = str.split("\n");
		var start = Math.max(lineno - 3, 0);
		var end = Math.min(lines.length, lineno + 3);
		var filename = esc(flnm);
		var context = lines.slice(start, end).map(function(line, i) {
			var curr = i + start + 1;
			return (curr == lineno ? " >> " : "    ") + curr + "| " + line;
		}).join("\n");
		err.path = filename;
		err.message = (filename || "ejs") + ":" + lineno + "\n" + context + "\n\n" + err.message;
		throw err;
	}
	function stripSemi(str) {
		return str.replace(/;(\s*$)/, "$1");
	}
	/**
	* Compile the given `str` of ejs into a template function.
	*
	* @param {String}  template EJS template
	*
	* @param {Options} [opts] compilation options
	*
	* @return {(TemplateFunction|ClientFunction)}
	* Depending on the value of `opts.client`, either type might be returned.
	* Note that the return type of the function also depends on the value of `opts.async`.
	* @public
	*/
	exports.compile = function compile(template, opts) {
		var templ;
		if (opts && opts.scope) {
			if (!scopeOptionWarned) {
				console.warn("`scope` option is deprecated and will be removed in EJS 3");
				scopeOptionWarned = true;
			}
			if (!opts.context) opts.context = opts.scope;
			delete opts.scope;
		}
		templ = new Template(template, opts);
		return templ.compile();
	};
	/**
	* Render the given `template` of ejs.
	*
	* If you would like to include options but not data, you need to explicitly
	* call this function with `data` being an empty object or `null`.
	*
	* @param {String}   template EJS template
	* @param {Object}  [data={}] template data
	* @param {Options} [opts={}] compilation and rendering options
	* @return {(String|Promise<String>)}
	* Return value type depends on `opts.async`.
	* @public
	*/
	exports.render = function(template, d$2, o$1) {
		var data = d$2 || utils.createNullProtoObjWherePossible();
		var opts = o$1 || utils.createNullProtoObjWherePossible();
		if (arguments.length == 2) utils.shallowCopyFromList(opts, data, _OPTS_PASSABLE_WITH_DATA);
		return handleCache(opts, template)(data);
	};
	/**
	* Render an EJS file at the given `path` and callback `cb(err, str)`.
	*
	* If you would like to include options but not data, you need to explicitly
	* call this function with `data` being an empty object or `null`.
	*
	* @param {String}             path     path to the EJS file
	* @param {Object}            [data={}] template data
	* @param {Options}           [opts={}] compilation and rendering options
	* @param {RenderFileCallback} cb callback
	* @public
	*/
	exports.renderFile = function() {
		var args = Array.prototype.slice.call(arguments);
		var filename = args.shift();
		var cb;
		var opts = { filename };
		var data;
		var viewOpts;
		if (typeof arguments[arguments.length - 1] == "function") cb = args.pop();
		if (args.length) {
			data = args.shift();
			if (args.length) utils.shallowCopy(opts, args.pop());
			else {
				if (data.settings) {
					if (data.settings.views) opts.views = data.settings.views;
					if (data.settings["view cache"]) opts.cache = true;
					viewOpts = data.settings["view options"];
					if (viewOpts) utils.shallowCopy(opts, viewOpts);
				}
				utils.shallowCopyFromList(opts, data, _OPTS_PASSABLE_WITH_DATA_EXPRESS);
			}
			opts.filename = filename;
		} else data = utils.createNullProtoObjWherePossible();
		return tryHandleCache(opts, data, cb);
	};
	/**
	* Clear intermediate JavaScript cache. Calls {@link Cache#reset}.
	* @public
	*/
	/**
	* EJS template class
	* @public
	*/
	exports.Template = Template;
	exports.clearCache = function() {
		exports.cache.reset();
	};
	function Template(text, optsParam) {
		var opts = utils.hasOwnOnlyObject(optsParam);
		var options = utils.createNullProtoObjWherePossible();
		this.templateText = text;
		/** @type {string | null} */
		this.mode = null;
		this.truncate = false;
		this.currentLine = 1;
		this.source = "";
		options.client = opts.client || false;
		options.escapeFunction = opts.escape || opts.escapeFunction || utils.escapeXML;
		options.compileDebug = opts.compileDebug !== false;
		options.debug = !!opts.debug;
		options.filename = opts.filename;
		options.openDelimiter = opts.openDelimiter || exports.openDelimiter || _DEFAULT_OPEN_DELIMITER;
		options.closeDelimiter = opts.closeDelimiter || exports.closeDelimiter || _DEFAULT_CLOSE_DELIMITER;
		options.delimiter = opts.delimiter || exports.delimiter || _DEFAULT_DELIMITER;
		options.strict = opts.strict || false;
		options.context = opts.context;
		options.cache = opts.cache || false;
		options.rmWhitespace = opts.rmWhitespace;
		options.root = opts.root;
		options.includer = opts.includer;
		options.outputFunctionName = opts.outputFunctionName;
		options.localsName = opts.localsName || exports.localsName || _DEFAULT_LOCALS_NAME;
		options.views = opts.views;
		options.async = opts.async;
		options.destructuredLocals = opts.destructuredLocals;
		options.legacyInclude = typeof opts.legacyInclude != "undefined" ? !!opts.legacyInclude : true;
		if (options.strict) options._with = false;
		else options._with = typeof opts._with != "undefined" ? opts._with : true;
		this.opts = options;
		this.regex = this.createRegex();
	}
	Template.modes = {
		EVAL: "eval",
		ESCAPED: "escaped",
		RAW: "raw",
		COMMENT: "comment",
		LITERAL: "literal"
	};
	Template.prototype = {
		createRegex: function() {
			var str = _REGEX_STRING;
			var delim = utils.escapeRegExpChars(this.opts.delimiter);
			var open = utils.escapeRegExpChars(this.opts.openDelimiter);
			var close = utils.escapeRegExpChars(this.opts.closeDelimiter);
			str = str.replace(/%/g, delim).replace(/</g, open).replace(/>/g, close);
			return new RegExp(str);
		},
		compile: function() {
			/** @type {string} */
			var src;
			/** @type {ClientFunction} */
			var fn;
			var opts = this.opts;
			var prepended = "";
			var appended = "";
			/** @type {EscapeCallback} */
			var escapeFn = opts.escapeFunction;
			/** @type {FunctionConstructor} */
			var ctor;
			/** @type {string} */
			var sanitizedFilename = opts.filename ? JSON.stringify(opts.filename) : "undefined";
			if (!this.source) {
				this.generateSource();
				prepended += "  var __output = \"\";\n  function __append(s) { if (s !== undefined && s !== null) __output += s }\n";
				if (opts.outputFunctionName) {
					if (!_JS_IDENTIFIER.test(opts.outputFunctionName)) throw new Error("outputFunctionName is not a valid JS identifier.");
					prepended += "  var " + opts.outputFunctionName + " = __append;\n";
				}
				if (opts.localsName && !_JS_IDENTIFIER.test(opts.localsName)) throw new Error("localsName is not a valid JS identifier.");
				if (opts.destructuredLocals && opts.destructuredLocals.length) {
					var destructuring = "  var __locals = (" + opts.localsName + " || {}),\n";
					for (var i = 0; i < opts.destructuredLocals.length; i++) {
						var name$2 = opts.destructuredLocals[i];
						if (!_JS_IDENTIFIER.test(name$2)) throw new Error("destructuredLocals[" + i + "] is not a valid JS identifier.");
						if (i > 0) destructuring += ",\n  ";
						destructuring += name$2 + " = __locals." + name$2;
					}
					prepended += destructuring + ";\n";
				}
				if (opts._with !== false) {
					prepended += "  with (" + opts.localsName + " || {}) {\n";
					appended += "  }\n";
				}
				appended += "  return __output;\n";
				this.source = prepended + this.source + appended;
			}
			if (opts.compileDebug) src = "var __line = 1\n  , __lines = " + JSON.stringify(this.templateText) + "\n  , __filename = " + sanitizedFilename + ";\ntry {\n" + this.source + "} catch (e) {\n  rethrow(e, __lines, __filename, __line, escapeFn);\n}\n";
			else src = this.source;
			if (opts.client) {
				src = "escapeFn = escapeFn || " + escapeFn.toString() + ";\n" + src;
				if (opts.compileDebug) src = "rethrow = rethrow || " + rethrow.toString() + ";\n" + src;
			}
			if (opts.strict) src = "\"use strict\";\n" + src;
			if (opts.debug) console.log(src);
			if (opts.compileDebug && opts.filename) src = src + "\n//# sourceURL=" + sanitizedFilename + "\n";
			try {
				if (opts.async) try {
					ctor = new Function("return (async function(){}).constructor;")();
				} catch (e) {
					if (e instanceof SyntaxError) throw new Error("This environment does not support async/await");
					else throw e;
				}
				else ctor = Function;
				fn = new ctor(opts.localsName + ", escapeFn, include, rethrow", src);
			} catch (e) {
				if (e instanceof SyntaxError) {
					if (opts.filename) e.message += " in " + opts.filename;
					e.message += " while compiling ejs\n\n";
					e.message += "If the above error is not helpful, you may want to try EJS-Lint:\n";
					e.message += "https://github.com/RyanZim/EJS-Lint";
					if (!opts.async) {
						e.message += "\n";
						e.message += "Or, if you meant to create an async function, pass `async: true` as an option.";
					}
				}
				throw e;
			}
			var returnedFn = opts.client ? fn : function anonymous(data) {
				var include = function(path$7, includeData) {
					var d$2 = utils.shallowCopy(utils.createNullProtoObjWherePossible(), data);
					if (includeData) d$2 = utils.shallowCopy(d$2, includeData);
					return includeFile(path$7, opts)(d$2);
				};
				return fn.apply(opts.context, [
					data || utils.createNullProtoObjWherePossible(),
					escapeFn,
					include,
					rethrow
				]);
			};
			if (opts.filename && typeof Object.defineProperty === "function") {
				var filename = opts.filename;
				var basename = path$6.basename(filename, path$6.extname(filename));
				try {
					Object.defineProperty(returnedFn, "name", {
						value: basename,
						writable: false,
						enumerable: false,
						configurable: true
					});
				} catch (e) {}
			}
			return returnedFn;
		},
		generateSource: function() {
			var opts = this.opts;
			if (opts.rmWhitespace) this.templateText = this.templateText.replace(/[\r\n]+/g, "\n").replace(/^\s+|\s+$/gm, "");
			this.templateText = this.templateText.replace(/[ \t]*<%_/gm, "<%_").replace(/_%>[ \t]*/gm, "_%>");
			var self = this;
			var matches = this.parseTemplateText();
			var d$2 = this.opts.delimiter;
			var o$1 = this.opts.openDelimiter;
			var c$1 = this.opts.closeDelimiter;
			if (matches && matches.length) matches.forEach(function(line, index) {
				var closing;
				if (line.indexOf(o$1 + d$2) === 0 && line.indexOf(o$1 + d$2 + d$2) !== 0) {
					closing = matches[index + 2];
					if (!(closing == d$2 + c$1 || closing == "-" + d$2 + c$1 || closing == "_" + d$2 + c$1)) throw new Error("Could not find matching close tag for \"" + line + "\".");
				}
				self.scanLine(line);
			});
		},
		parseTemplateText: function() {
			var str = this.templateText;
			var pat = this.regex;
			var result = pat.exec(str);
			var arr = [];
			var firstPos;
			while (result) {
				firstPos = result.index;
				if (firstPos !== 0) {
					arr.push(str.substring(0, firstPos));
					str = str.slice(firstPos);
				}
				arr.push(result[0]);
				str = str.slice(result[0].length);
				result = pat.exec(str);
			}
			if (str) arr.push(str);
			return arr;
		},
		_addOutput: function(line) {
			if (this.truncate) {
				line = line.replace(/^(?:\r\n|\r|\n)/, "");
				this.truncate = false;
			}
			if (!line) return line;
			line = line.replace(/\\/g, "\\\\");
			line = line.replace(/\n/g, "\\n");
			line = line.replace(/\r/g, "\\r");
			line = line.replace(/"/g, "\\\"");
			this.source += "    ; __append(\"" + line + "\")\n";
		},
		scanLine: function(line) {
			var self = this;
			var d$2 = this.opts.delimiter;
			var o$1 = this.opts.openDelimiter;
			var c$1 = this.opts.closeDelimiter;
			var newLineCount = 0;
			newLineCount = line.split("\n").length - 1;
			switch (line) {
				case o$1 + d$2:
				case o$1 + d$2 + "_":
					this.mode = Template.modes.EVAL;
					break;
				case o$1 + d$2 + "=":
					this.mode = Template.modes.ESCAPED;
					break;
				case o$1 + d$2 + "-":
					this.mode = Template.modes.RAW;
					break;
				case o$1 + d$2 + "#":
					this.mode = Template.modes.COMMENT;
					break;
				case o$1 + d$2 + d$2:
					this.mode = Template.modes.LITERAL;
					this.source += "    ; __append(\"" + line.replace(o$1 + d$2 + d$2, o$1 + d$2) + "\")\n";
					break;
				case d$2 + d$2 + c$1:
					this.mode = Template.modes.LITERAL;
					this.source += "    ; __append(\"" + line.replace(d$2 + d$2 + c$1, d$2 + c$1) + "\")\n";
					break;
				case d$2 + c$1:
				case "-" + d$2 + c$1:
				case "_" + d$2 + c$1:
					if (this.mode == Template.modes.LITERAL) this._addOutput(line);
					this.mode = null;
					this.truncate = line.indexOf("-") === 0 || line.indexOf("_") === 0;
					break;
				default: if (this.mode) {
					switch (this.mode) {
						case Template.modes.EVAL:
						case Template.modes.ESCAPED:
						case Template.modes.RAW: if (line.lastIndexOf("//") > line.lastIndexOf("\n")) line += "\n";
					}
					switch (this.mode) {
						case Template.modes.EVAL:
							this.source += "    ; " + line + "\n";
							break;
						case Template.modes.ESCAPED:
							this.source += "    ; __append(escapeFn(" + stripSemi(line) + "))\n";
							break;
						case Template.modes.RAW:
							this.source += "    ; __append(" + stripSemi(line) + ")\n";
							break;
						case Template.modes.COMMENT: break;
						case Template.modes.LITERAL:
							this._addOutput(line);
							break;
					}
				} else this._addOutput(line);
			}
			if (self.opts.compileDebug && newLineCount) {
				this.currentLine += newLineCount;
				this.source += "    ; __line = " + this.currentLine + "\n";
			}
		}
	};
	/**
	* Escape characters reserved in XML.
	*
	* This is simply an export of {@link module:utils.escapeXML}.
	*
	* If `markup` is `undefined` or `null`, the empty string is returned.
	*
	* @param {String} markup Input string
	* @return {String} Escaped string
	* @public
	* @func
	* */
	exports.escapeXML = utils.escapeXML;
	/**
	* Express.js support.
	*
	* This is an alias for {@link module:ejs.renderFile}, in order to support
	* Express.js out-of-the-box.
	*
	* @func
	*/
	exports.__express = exports.renderFile;
	/**
	* Version of EJS.
	*
	* @readonly
	* @type {String}
	* @public
	*/
	exports.VERSION = _VERSION_STRING;
	/**
	* Name for detection of EJS.
	*
	* @readonly
	* @type {String}
	* @public
	*/
	exports.name = _NAME;
	if (typeof window != "undefined") window.ejs = exports;
} });
var import_ejs = __toESM(require_ejs(), 1);

//#endregion
//#region utils/banners.ts
const defaultBanner = "Vue.js - The Progressive JavaScript Framework";
const gradientBanner = "\x1B[38;2;66;211;146mV\x1B[39m\x1B[38;2;66;211;146mu\x1B[39m\x1B[38;2;66;211;146me\x1B[39m\x1B[38;2;66;211;146m.\x1B[39m\x1B[38;2;66;211;146mj\x1B[39m\x1B[38;2;67;209;149ms\x1B[39m \x1B[38;2;68;206;152m-\x1B[39m \x1B[38;2;69;204;155mT\x1B[39m\x1B[38;2;70;201;158mh\x1B[39m\x1B[38;2;71;199;162me\x1B[39m \x1B[38;2;72;196;165mP\x1B[39m\x1B[38;2;73;194;168mr\x1B[39m\x1B[38;2;74;192;171mo\x1B[39m\x1B[38;2;75;189;174mg\x1B[39m\x1B[38;2;76;187;177mr\x1B[39m\x1B[38;2;77;184;180me\x1B[39m\x1B[38;2;78;182;183ms\x1B[39m\x1B[38;2;79;179;186ms\x1B[39m\x1B[38;2;80;177;190mi\x1B[39m\x1B[38;2;81;175;193mv\x1B[39m\x1B[38;2;82;172;196me\x1B[39m \x1B[38;2;83;170;199mJ\x1B[39m\x1B[38;2;83;167;202ma\x1B[39m\x1B[38;2;84;165;205mv\x1B[39m\x1B[38;2;85;162;208ma\x1B[39m\x1B[38;2;86;160;211mS\x1B[39m\x1B[38;2;87;158;215mc\x1B[39m\x1B[38;2;88;155;218mr\x1B[39m\x1B[38;2;89;153;221mi\x1B[39m\x1B[38;2;90;150;224mp\x1B[39m\x1B[38;2;91;148;227mt\x1B[39m \x1B[38;2;92;145;230mF\x1B[39m\x1B[38;2;93;143;233mr\x1B[39m\x1B[38;2;94;141;236ma\x1B[39m\x1B[38;2;95;138;239mm\x1B[39m\x1B[38;2;96;136;243me\x1B[39m\x1B[38;2;97;133;246mw\x1B[39m\x1B[38;2;98;131;249mo\x1B[39m\x1B[38;2;99;128;252mr\x1B[39m\x1B[38;2;100;126;255mk\x1B[39m";

//#endregion
//#region utils/deepMerge.ts
const isObject$1 = (val) => val && typeof val === "object";
const mergeArrayWithDedupe$1 = (a, b$1) => Array.from(new Set([...a, ...b$1]));
/**
* Recursively merge the content of the new object to the existing one
* @param {Object} target the existing object
* @param {Object} obj the new object
*/
function deepMerge$1(target, obj) {
	for (const key of Object.keys(obj)) {
		const oldVal = target[key];
		const newVal = obj[key];
		if (Array.isArray(oldVal) && Array.isArray(newVal)) target[key] = mergeArrayWithDedupe$1(oldVal, newVal);
		else if (isObject$1(oldVal) && isObject$1(newVal)) target[key] = deepMerge$1(oldVal, newVal);
		else target[key] = newVal;
	}
	return target;
}
var deepMerge_default = deepMerge$1;

//#endregion
//#region utils/sortDependencies.ts
function sortDependencies(packageJson) {
	const sorted = {};
	const depTypes = [
		"dependencies",
		"devDependencies",
		"peerDependencies",
		"optionalDependencies"
	];
	for (const depType of depTypes) if (packageJson[depType]) {
		sorted[depType] = {};
		Object.keys(packageJson[depType]).sort().forEach((name$2) => {
			sorted[depType][name$2] = packageJson[depType][name$2];
		});
	}
	return {
		...packageJson,
		...sorted
	};
}

//#endregion
//#region utils/renderTemplate.ts
/**
* Renders a template folder/file to the file system,
* by recursively copying all files under the `src` directory,
* with the following exception:
*   - `_filename` should be renamed to `.filename`
*   - Fields in `package.json` should be recursively merged
* @param {string} src source filename to copy
* @param {string} dest destination filename of the copy operation
*/
function renderTemplate(src, dest, callbacks) {
	const stats = fs$5.statSync(src);
	if (stats.isDirectory()) {
		if (path$5.basename(src) === "node_modules") return;
		fs$5.mkdirSync(dest, { recursive: true });
		for (const file of fs$5.readdirSync(src)) renderTemplate(path$5.resolve(src, file), path$5.resolve(dest, file), callbacks);
		return;
	}
	const filename = path$5.basename(src);
	if (filename === "package.json" && fs$5.existsSync(dest)) {
		const existing = JSON.parse(fs$5.readFileSync(dest, "utf8"));
		const newPackage = JSON.parse(fs$5.readFileSync(src, "utf8"));
		const pkg = sortDependencies(deepMerge_default(existing, newPackage));
		fs$5.writeFileSync(dest, JSON.stringify(pkg, null, 2) + "\n");
		return;
	}
	if (filename === "extensions.json" && fs$5.existsSync(dest)) {
		const existing = JSON.parse(fs$5.readFileSync(dest, "utf8"));
		const newExtensions = JSON.parse(fs$5.readFileSync(src, "utf8"));
		const extensions = deepMerge_default(existing, newExtensions);
		fs$5.writeFileSync(dest, JSON.stringify(extensions, null, 2) + "\n");
		return;
	}
	if (filename === "settings.json" && fs$5.existsSync(dest)) {
		const existing = JSON.parse(fs$5.readFileSync(dest, "utf8"));
		const newSettings = JSON.parse(fs$5.readFileSync(src, "utf8"));
		const settings = deepMerge_default(existing, newSettings);
		fs$5.writeFileSync(dest, JSON.stringify(settings, null, 2) + "\n");
		return;
	}
	if (filename.startsWith("_")) dest = path$5.resolve(path$5.dirname(dest), filename.replace(/^_/, "."));
	if (filename === "_gitignore" && fs$5.existsSync(dest)) {
		const existing = fs$5.readFileSync(dest, "utf8");
		const newGitignore = fs$5.readFileSync(src, "utf8");
		fs$5.writeFileSync(dest, existing + "\n" + newGitignore);
		return;
	}
	if (filename.endsWith(".data.mjs")) {
		dest = dest.replace(/\.data\.mjs$/, "");
		callbacks.push(async (dataStore) => {
			const getData = (await import(pathToFileURL(src).toString())).default;
			dataStore[dest] = await getData({ oldData: dataStore[dest] || {} });
		});
		return;
	}
	fs$5.copyFileSync(src, dest);
}
var renderTemplate_default = renderTemplate;

//#endregion
//#region utils/directoryTraverse.ts
function preOrderDirectoryTraverse(dir, dirCallback, fileCallback) {
	for (const filename of fs$4.readdirSync(dir)) {
		if (filename === ".git") continue;
		const fullpath = path$4.resolve(dir, filename);
		if (fs$4.lstatSync(fullpath).isDirectory()) {
			dirCallback(fullpath);
			if (fs$4.existsSync(fullpath)) preOrderDirectoryTraverse(fullpath, dirCallback, fileCallback);
			continue;
		}
		fileCallback(fullpath);
	}
}
function postOrderDirectoryTraverse(dir, dirCallback, fileCallback) {
	for (const filename of fs$4.readdirSync(dir)) {
		if (filename === ".git") continue;
		const fullpath = path$4.resolve(dir, filename);
		if (fs$4.lstatSync(fullpath).isDirectory()) {
			postOrderDirectoryTraverse(fullpath, dirCallback, fileCallback);
			dirCallback(fullpath);
			continue;
		}
		fileCallback(fullpath);
	}
}

//#endregion
//#region utils/getCommand.ts
function getCommand(packageManager$1, scriptName, args) {
	if (scriptName === "install") return packageManager$1 === "yarn" ? "yarn" : `${packageManager$1} install`;
	if (scriptName === "build") return packageManager$1 === "npm" || packageManager$1 === "bun" ? `${packageManager$1} run build` : `${packageManager$1} build`;
	if (args) return packageManager$1 === "npm" ? `npm run ${scriptName} -- ${args}` : `${packageManager$1} ${scriptName} ${args}`;
	else return packageManager$1 === "npm" ? `npm run ${scriptName}` : `${packageManager$1} ${scriptName}`;
}

//#endregion
//#region utils/generateReadme.ts
const sfcTypeSupportDoc = [
	"",
	"## Type Support for `.vue` Imports in TS",
	"",
	"TypeScript cannot handle type information for `.vue` imports by default, so we replace the `tsc` CLI with `vue-tsc` for type checking. In editors, we need [Volar](https://marketplace.visualstudio.com/items?itemName=Vue.volar) to make the TypeScript language service aware of `.vue` types.",
	""
].join("\n");
function generateReadme({ projectName, packageManager: packageManager$1, needsTypeScript, needsCypress, needsNightwatch, needsCypressCT, needsNightwatchCT, needsPlaywright, needsVitest, needsEslint }) {
	const commandFor = (scriptName, args) => getCommand(packageManager$1, scriptName, args);
	let readme = `# ${projectName}

This template should help get you started developing with Vue 3 in Vite.

## Recommended IDE Setup

[VSCode](https://code.visualstudio.com/) + [Volar](https://marketplace.visualstudio.com/items?itemName=Vue.volar) (and disable Vetur).
${needsTypeScript ? sfcTypeSupportDoc : ""}
## Customize configuration

See [Vite Configuration Reference](https://vite.dev/config/).

## Project Setup

`;
	let installCommand = commandFor("install");
	if (packageManager$1 === "pnpm" && needsNightwatch) installCommand += `\npnpm approve-builds # for pnpm 10+`;
	let npmScriptsDescriptions = `\`\`\`sh
${installCommand}
\`\`\`

### Compile and Hot-Reload for Development

\`\`\`sh
${commandFor("dev")}
\`\`\`

### ${needsTypeScript ? "Type-Check, " : ""}Compile and Minify for Production

\`\`\`sh
${commandFor("build")}
\`\`\`
`;
	if (needsVitest) npmScriptsDescriptions += `
### Run Unit Tests with [Vitest](https://vitest.dev/)

\`\`\`sh
${commandFor("test:unit")}
\`\`\`
`;
	if (needsCypressCT) npmScriptsDescriptions += `
### Run Headed Component Tests with [Cypress Component Testing](https://on.cypress.io/component)

\`\`\`sh
${commandFor("test:unit:dev")} # or \`${commandFor("test:unit")}\` for headless testing
\`\`\`
`;
	if (needsCypress) npmScriptsDescriptions += `
### Run End-to-End Tests with [Cypress](https://www.cypress.io/)

\`\`\`sh
${commandFor("test:e2e:dev")}
\`\`\`

This runs the end-to-end tests against the Vite development server.
It is much faster than the production build.

But it's still recommended to test the production build with \`test:e2e\` before deploying (e.g. in CI environments):

\`\`\`sh
${commandFor("build")}
${commandFor("test:e2e")}
\`\`\`
`;
	if (needsNightwatch) npmScriptsDescriptions += `
### Run End-to-End Tests with [Nightwatch](https://nightwatchjs.org/)

\`\`\`sh
# When using CI, the project must be built first.
${commandFor("build")}

# Runs the end-to-end tests
${commandFor("test:e2e")}
# Runs the tests only on Chrome
${commandFor("test:e2e", "--env chrome")}
# Runs the tests of a specific file
${commandFor("test:e2e", `tests/e2e/example.${needsTypeScript ? "ts" : "js"}`)}
# Runs the tests in debug mode
${commandFor("test:e2e", "--debug")}
\`\`\`
    `;
	if (needsNightwatchCT) npmScriptsDescriptions += `
### Run Headed Component Tests with [Nightwatch Component Testing](https://nightwatchjs.org/guide/component-testing/introduction.html)
  
\`\`\`sh
${commandFor("test:unit")}
${commandFor("test:unit -- --headless # for headless testing")}
\`\`\`
`;
	if (needsPlaywright) npmScriptsDescriptions += `
### Run End-to-End Tests with [Playwright](https://playwright.dev)

\`\`\`sh
# Install browsers for the first run
npx playwright install

# When testing on CI, must build the project first
${commandFor("build")}

# Runs the end-to-end tests
${commandFor("test:e2e")}
# Runs the tests only on Chromium
${commandFor("test:e2e", "--project=chromium")}
# Runs the tests of a specific file
${commandFor("test:e2e", "tests/example.spec.ts")}
# Runs the tests in debug mode
${commandFor("test:e2e", "--debug")}
\`\`\`
`;
	if (needsEslint) npmScriptsDescriptions += `
### Lint with [ESLint](https://eslint.org/)

\`\`\`sh
${commandFor("lint")}
\`\`\`
`;
	readme += npmScriptsDescriptions;
	return readme;
}

//#endregion
//#region utils/getLanguage.ts
/**
*
* This function is used to link obtained locale with correct locale file in order to make locales reusable
*
* @param locale the obtained locale
* @returns locale that linked with correct name
*/
function linkLocale(locale) {
	if (locale === "C") return "en-US";
	let linkedLocale;
	try {
		linkedLocale = Intl.getCanonicalLocales(locale)[0];
	} catch (error) {
		console.log(`${error.toString()}, invalid language tag: "${locale}"\n`);
	}
	switch (linkedLocale) {
		case "zh-TW":
		case "zh-HK":
		case "zh-MO":
			linkedLocale = "zh-Hant";
			break;
		case "zh-CN":
		case "zh-SG":
			linkedLocale = "zh-Hans";
			break;
		default: linkedLocale = locale;
	}
	return linkedLocale;
}
function getLocale() {
	const shellLocale = process.env.LC_ALL || process.env.LC_MESSAGES || process.env.LANG || Intl.DateTimeFormat().resolvedOptions().locale || "en-US";
	return linkLocale(shellLocale.split(".")[0].replace("_", "-"));
}
async function loadLanguageFile(filePath) {
	return await fs$3.promises.readFile(filePath, "utf-8").then((data) => {
		const parsedData = JSON.parse(data);
		if (parsedData) return parsedData;
	});
}
async function getLanguage(localesRoot) {
	const locale = getLocale();
	const languageFilePath = path$3.resolve(localesRoot, `${locale}.json`);
	const fallbackPath = path$3.resolve(localesRoot, "en-US.json");
	const doesLanguageExist = fs$3.existsSync(languageFilePath);
	const lang = doesLanguageExist ? await loadLanguageFile(languageFilePath) : await loadLanguageFile(fallbackPath);
	return lang;
}

//#endregion
//#region node_modules/.pnpm/@vue+create-eslint-config@0.10.1/node_modules/@vue/create-eslint-config/renderEjsFile.js
const templates = {
	"./templates/_editorconfig.ejs": "[*.{js,jsx,mjs,cjs,ts,tsx,mts,cts,vue,css,scss,sass,less,styl}]\ncharset = utf-8\nindent_size = 2\nindent_style = space\ninsert_final_newline = true\ntrim_trailing_whitespace = true\n<%_ if (styleGuide !== 'standard' || needsPrettier) { _%>\n<%#\n// standard doesn't have an opinion on line endings\n// https://github.com/standard/standard/issues/140\n// or maximum line length\n// https://github.com/standard/standard/issues/1559\n// Prettier enforces these things, though.\n%>\nend_of_line = lf\nmax_line_length = 100\n<%_ } _%>\n",
	"./templates/_gitattributes": "* text=auto eol=lf\n",
	"./templates/_prettierrc.json.ejs": "<%_ if (styleGuide === 'airbnb') { _%>\n{\n  \"$schema\": \"https://json.schemastore.org/prettierrc\",\n  \"printWidth\": 100,\n  \"singleQuote\": true\n}\n<%_ } else if (styleGuide === 'standard') { _%>\n{\n  \"$schema\": \"https://json.schemastore.org/prettierrc\",\n  \"semi\": false,\n  \"singleQuote\": true,\n  \"jsxSingleQuote\": true,\n  \"trailingComma\": \"none\"\n}\n<%_ } else { _%>\n<%#\n  The default style follows the one used in the vuejs/create-vue repository\n  <https://github.com/vuejs/create-vue/blob/main/.prettierrc>\n%>\n{\n  \"$schema\": \"https://json.schemastore.org/prettierrc\",\n  \"semi\": false,\n  \"singleQuote\": true,\n  \"printWidth\": 100\n}\n<%_ } _%>\n",
	"./templates/eslint.config.js.ejs": "<%_ for (const { importer } of configsBeforeVuePlugin) { _%>\n<%- importer %>\n<%_ } _%>\nimport { defineConfig, globalIgnores } from 'eslint/config'\nimport globals from 'globals'\nimport js from '@eslint/js'\nimport pluginVue from 'eslint-plugin-vue'\n<%_ for (const { importer } of configsAfterVuePlugin) { _%>\n<%- importer %>\n<%_ } _%>\n\nexport default defineConfig([\n  {\n    name: 'app/files-to-lint',\n    files: ['**/*.{<%= fileExtensions.join(',') %>}'],\n  },\n\n  globalIgnores(['**/dist/**', '**/dist-ssr/**', '**/coverage/**']),\n\n  {\n    languageOptions: {\n      globals: {\n        ...globals.browser,\n      },\n    },\n  },\n\n<%_ for (const { content } of configsBeforeVuePlugin) { _%>\n  <%- content %><%# TODO: auto-indent if content's multi-line %>\n<%_ } _%>\n  js.configs.recommended,\n  ...pluginVue.configs['flat/essential'],\n<%_ for (const { content } of configsAfterVuePlugin) { _%>\n  <%- content %><%# TODO: auto-indent if content's multi-line %>\n<%_ } _%>\n])\n",
	"./templates/eslint.config.ts.ejs": "<%_ for (const { importer } of configsBeforeVuePlugin) { _%>\n<%- importer %>\n<%_ } _%>\nimport { globalIgnores } from 'eslint/config'\nimport { defineConfigWithVueTs, vueTsConfigs } from '@vue/eslint-config-typescript'\nimport pluginVue from 'eslint-plugin-vue'\n<%_ for (const { importer } of configsAfterVuePlugin) { _%>\n<%- importer %>\n<%_ } _%>\n\n// To allow more languages other than `ts` in `.vue` files, uncomment the following lines:\n// import { configureVueProject } from '@vue/eslint-config-typescript'\n// configureVueProject({ scriptLangs: ['ts', 'tsx'] })\n// More info at https://github.com/vuejs/eslint-config-typescript/#advanced-setup\n\nexport default defineConfigWithVueTs(\n  {\n    name: 'app/files-to-lint',\n    files: ['**/*.{<%= fileExtensions.join(',') %>}'],\n  },\n\n  globalIgnores(['**/dist/**', '**/dist-ssr/**', '**/coverage/**']),\n\n<%_ for (const { content } of configsBeforeVuePlugin) { _%>\n  <%- content %><%# TODO: auto-indent if content's multi-line %>\n<%_ } _%>\n  pluginVue.configs['flat/essential'],\n  vueTsConfigs.recommended,\n<%_ for (const { content } of configsAfterVuePlugin) { _%>\n  <%- content %><%# TODO: auto-indent if content's multi-line %>\n<%_ } _%>\n)\n"
};
function renderEjsFile(filePath, data) {
	return import_ejs.render(templates[filePath], data, {});
}

//#endregion
//#region node_modules/.pnpm/@vue+create-eslint-config@0.10.1/node_modules/@vue/create-eslint-config/package.json
var name$1 = "@vue/create-eslint-config";
var version$1 = "0.10.1";
var description$1 = "Utility to setup ESLint in Vue.js projects.";
var type$1 = "module";
var main = "index.js";
var bin$1 = { "create-eslint-config": "bin/create-eslint-config.js" };
var engines$1 = { "node": "^16.14.0 || >= 18.0.0" };
var scripts$1 = {
	"test": "echo \"Error: no test specified\" && exit 1",
	"lint": "eslint --fix ."
};
var repository$1 = {
	"type": "git",
	"url": "git+https://github.com/vuejs/create-eslint-config.git"
};
var keywords$1 = [
	"vue",
	"eslint",
	"config"
];
var author$1 = "Haoqun Jiang <haoqunjiang+npm@gmail.com>";
var license$1 = "MIT";
var bugs$1 = { "url": "https://github.com/vuejs/create-eslint-config/issues" };
var homepage$1 = "https://github.com/vuejs/create-eslint-config#readme";
var publishConfig$1 = {
	"access": "public",
	"provenance": true
};
var dependencies = {
	"ejs": "^3.1.10",
	"enquirer": "^2.4.1",
	"kolorist": "^1.8.0"
};
var devDependencies$2 = {
	"@eslint/js": "^9.22.0",
	"@types/node": "^22.13.10",
	"@vue/eslint-config-prettier": "^10.2.0",
	"@vue/eslint-config-typescript": "^14.5.0",
	"eslint": "^9.22.0",
	"eslint-plugin-oxlint": "^0.16.0",
	"eslint-plugin-vue": "~10.0.0",
	"globals": "^16.0.0",
	"jiti": "^2.4.2",
	"npm-run-all2": "^7.0.2",
	"oxlint": "^0.16.0",
	"prettier": "3.5.3",
	"typescript": "~5.8.2"
};
var package_default$2 = {
	name: name$1,
	version: version$1,
	description: description$1,
	type: type$1,
	main,
	bin: bin$1,
	engines: engines$1,
	scripts: scripts$1,
	repository: repository$1,
	keywords: keywords$1,
	author: author$1,
	license: license$1,
	bugs: bugs$1,
	homepage: homepage$1,
	publishConfig: publishConfig$1,
	dependencies,
	devDependencies: devDependencies$2
};

//#endregion
//#region node_modules/.pnpm/@vue+create-eslint-config@0.10.1/node_modules/@vue/create-eslint-config/index.js
const versionMap = package_default$2.devDependencies;
function createConfig({ styleGuide = "default", hasTypeScript = false, needsPrettier = false, needsOxlint = false, additionalConfigs = [] }) {
	const pickDependencies = (keys) => pickKeysFromObject(versionMap, keys);
	const pkg = {
		devDependencies: pickDependencies(["eslint", "eslint-plugin-vue"]),
		scripts: {}
	};
	const fileExtensions = ["vue"];
	if (hasTypeScript) {
		fileExtensions.unshift("ts", "mts", "tsx");
		additionalConfigs.unshift({ devDependencies: pickDependencies(["@vue/eslint-config-typescript", "jiti"]) });
	} else {
		fileExtensions.unshift("js", "mjs", "jsx");
		additionalConfigs.unshift({ devDependencies: pickDependencies(["@eslint/js", "globals"]) });
	}
	if (needsOxlint) {
		additionalConfigs.push({
			devDependencies: pickDependencies([
				"oxlint",
				"eslint-plugin-oxlint",
				"npm-run-all2"
			]),
			afterVuePlugin: [{
				importer: "import pluginOxlint from 'eslint-plugin-oxlint'",
				content: "...pluginOxlint.configs['flat/recommended'],"
			}]
		});
		pkg.scripts["lint:oxlint"] = "oxlint . --fix -D correctness --ignore-path .gitignore";
		pkg.scripts["lint:eslint"] = "eslint . --fix";
		pkg.scripts.lint = "run-s lint:*";
	} else pkg.scripts.lint = "eslint . --fix";
	if (needsPrettier) {
		additionalConfigs.push({
			devDependencies: pickDependencies(["prettier", "@vue/eslint-config-prettier"]),
			afterVuePlugin: [{
				importer: "import skipFormatting from '@vue/eslint-config-prettier/skip-formatting'",
				content: "skipFormatting,"
			}]
		});
		pkg.scripts.format = "prettier --write src/";
	}
	const configsBeforeVuePlugin = [], configsAfterVuePlugin = [];
	for (const config of additionalConfigs) {
		deepMerge(pkg.devDependencies, config.devDependencies ?? {});
		configsBeforeVuePlugin.push(...config.beforeVuePlugin ?? []);
		configsAfterVuePlugin.push(...config.afterVuePlugin ?? []);
	}
	const templateData = {
		styleGuide,
		needsPrettier,
		fileExtensions,
		configsBeforeVuePlugin,
		configsAfterVuePlugin
	};
	const files$1 = { ".editorconfig": renderEjsFile("./templates/_editorconfig.ejs", templateData) };
	if (hasTypeScript) files$1["eslint.config.ts"] = renderEjsFile("./templates/eslint.config.ts.ejs", templateData);
	else files$1["eslint.config.js"] = renderEjsFile("./templates/eslint.config.js.ejs", templateData);
	if (needsPrettier) files$1[".prettierrc.json"] = renderEjsFile("./templates/_prettierrc.json.ejs", templateData);
	if (styleGuide !== "standard" || needsPrettier) files$1[".gitattributes"] = renderEjsFile("./templates/_gitattributes", {});
	return {
		pkg,
		files: files$1
	};
}
/**
* Picks specified keys from an object.
*
* @param {Object} obj - The source object.
* @param {string[]} keys - The keys to pick from the object.
* @returns {Object} - A new object with the picked keys.
*/
function pickKeysFromObject(obj, keys) {
	return keys.reduce((acc, key) => {
		if (key in obj) acc[key] = obj[key];
		return acc;
	}, {});
}
const isObject = (val) => val && typeof val === "object";
const mergeArrayWithDedupe = (a, b$1) => Array.from(new Set([...a, ...b$1]));
function deepMerge(target, obj) {
	for (const key of Object.keys(obj)) {
		const oldVal = target[key];
		const newVal = obj[key];
		if (Array.isArray(oldVal) && Array.isArray(newVal)) target[key] = mergeArrayWithDedupe(oldVal, newVal);
		else if (isObject(oldVal) && isObject(newVal)) target[key] = deepMerge(oldVal, newVal);
		else target[key] = newVal;
	}
	return target;
}

//#endregion
//#region template/eslint/package.json
var devDependencies$1 = {
	"@vitest/eslint-plugin": "^1.1.39",
	"eslint-plugin-cypress": "^4.2.0",
	"eslint-plugin-playwright": "^2.2.0"
};
var package_default$1 = { devDependencies: devDependencies$1 };

//#endregion
//#region utils/renderEslint.ts
const eslintDeps = package_default$1.devDependencies;
function renderEslint(rootDir, { needsTypeScript, needsVitest, needsCypress, needsCypressCT, needsOxlint, needsPrettier, needsPlaywright }) {
	const additionalConfigs = getAdditionalConfigs({
		needsTypeScript,
		needsVitest,
		needsCypress,
		needsCypressCT,
		needsPlaywright
	});
	const { pkg, files: files$1 } = createConfig({
		styleGuide: "default",
		hasTypeScript: needsTypeScript,
		needsOxlint,
		needsPrettier,
		additionalConfigs
	});
	const packageJsonPath = path$2.resolve(rootDir, "package.json");
	const existingPkg = JSON.parse(fs$2.readFileSync(packageJsonPath, "utf8"));
	const updatedPkg = sortDependencies(deepMerge_default(existingPkg, pkg));
	fs$2.writeFileSync(packageJsonPath, JSON.stringify(updatedPkg, null, 2) + "\n", "utf8");
	for (const [fileName, content] of Object.entries(files$1)) {
		const fullPath = path$2.resolve(rootDir, fileName);
		fs$2.writeFileSync(fullPath, content, "utf8");
	}
}
function getAdditionalConfigs({ needsTypeScript, needsVitest, needsCypress, needsCypressCT, needsPlaywright }) {
	const additionalConfigs = [];
	if (needsVitest) additionalConfigs.push({
		devDependencies: { "@vitest/eslint-plugin": eslintDeps["@vitest/eslint-plugin"] },
		afterVuePlugin: [{
			importer: `import pluginVitest from '@vitest/eslint-plugin'`,
			content: `
  {
    ...pluginVitest.configs.recommended,
    files: ['src/**/__tests__/*'],
  },`
		}]
	});
	if (needsCypress) additionalConfigs.push({
		devDependencies: { "eslint-plugin-cypress": eslintDeps["eslint-plugin-cypress"] },
		afterVuePlugin: [{
			importer: (needsTypeScript ? "// eslint-disable-next-line @typescript-eslint/ban-ts-comment\n// @ts-ignore\n" : "") + "import pluginCypress from 'eslint-plugin-cypress/flat'",
			content: `
  {
    ...pluginCypress.configs.recommended,
    files: [
      ${[
				...needsCypressCT ? ["**/__tests__/*.{cy,spec}.{js,ts,jsx,tsx}"] : [],
				"cypress/e2e/**/*.{cy,spec}.{js,ts,jsx,tsx}",
				"cypress/support/**/*.{js,ts,jsx,tsx}"
			].map(JSON.stringify.bind(JSON)).join(",\n      ").replace(
				/"/g,
				"'"
				/* use single quotes as in the other configs */
)}
    ],
  },`
		}]
	});
	if (needsPlaywright) additionalConfigs.push({
		devDependencies: { "eslint-plugin-playwright": eslintDeps["eslint-plugin-playwright"] },
		afterVuePlugin: [{
			importer: "import pluginPlaywright from 'eslint-plugin-playwright'",
			content: `
  {
    ...pluginPlaywright.configs['flat/recommended'],
    files: ['e2e/**/*.{test,spec}.{js,ts,jsx,tsx}'],
  },`
		}]
	});
	return additionalConfigs;
}

//#endregion
//#region utils/trimBoilerplate.ts
function replaceContent(filepath, replacer) {
	const content = fs$1.readFileSync(filepath, "utf8");
	fs$1.writeFileSync(filepath, replacer(content));
}
function trimBoilerplate(rootDir) {
	const srcDir = path$1.resolve(rootDir, "src");
	for (const filename of fs$1.readdirSync(srcDir)) {
		if ([
			"main.js",
			"main.ts",
			"router",
			"stores"
		].includes(filename)) continue;
		const fullpath = path$1.resolve(srcDir, filename);
		fs$1.rmSync(fullpath, { recursive: true });
	}
}
function removeCSSImport(rootDir, needsTypeScript, needsCypressCT) {
	const entryPath = path$1.resolve(rootDir, needsTypeScript ? "src/main.ts" : "src/main.js");
	replaceContent(entryPath, (content) => content.replace("import './assets/main.css'\n\n", ""));
	if (needsCypressCT) {
		const ctSetupPath = path$1.resolve(rootDir, needsTypeScript ? "cypress/support/component.ts" : "cypress/support/component.js");
		replaceContent(ctSetupPath, (content) => content.replace("import '@/assets/main.css'", "// import '@/assets/main.css'"));
	}
}
function emptyRouterConfig(rootDir, needsTypeScript) {
	const srcDir = path$1.resolve(rootDir, "src");
	const routerEntry = path$1.resolve(srcDir, needsTypeScript ? "router/index.ts" : "router/index.js");
	replaceContent(routerEntry, (content) => content.replace(`import HomeView from '../views/HomeView.vue'\n`, "").replace(/routes:\s*\[[\s\S]*?\],/, "routes: [],"));
}

//#endregion
//#region package.json
var name = "create-vue";
var version = "3.16.4";
var description = "🛠️ The recommended way to start a Vite-powered Vue project";
var type = "module";
var packageManager = "pnpm@10.7.1";
var bin = { "create-vue": "bundle.js" };
var files = [
	"locales",
	"bundle.js",
	"template",
	"!template/**/node_modules/.bin/*"
];
var engines = { "node": ">=v18.3.0" };
var scripts = {
	"prepare": "husky",
	"format": "prettier --write .",
	"build": "rolldown -c rolldown.config.ts",
	"snapshot": "zx ./scripts/snapshot.mjs",
	"pretest": "pnpm run build && pnpm run snapshot",
	"test": "zx ./scripts/test.mjs",
	"test:unit": "vitest",
	"preversion": "git fetch && git status -uno | grep -q 'Your branch is up to date'",
	"postversion": "zx ./scripts/postversion.mjs",
	"prepublishOnly": "pnpm run build"
};
var repository = {
	"type": "git",
	"url": "git+https://github.com/vuejs/create-vue.git"
};
var keywords = [];
var author = "Haoqun Jiang <npm@haoqun.me>";
var license = "MIT";
var bugs = { "url": "https://github.com/vuejs/create-vue/issues" };
var homepage = "https://github.com/vuejs/create-vue#readme";
var devDependencies = {
	"@clack/prompts": "^0.10.0",
	"@tsconfig/node22": "^22.0.1",
	"@types/eslint": "^9.6.1",
	"@types/node": "^22.14.0",
	"@types/prompts": "^2.4.9",
	"@vue/create-eslint-config": "^0.10.1",
	"@vue/tsconfig": "^0.7.0",
	"ejs": "^3.1.10",
	"husky": "^9.1.7",
	"lint-staged": "^15.5.0",
	"picocolors": "^1.1.1",
	"prettier": "3.5.3",
	"rolldown": "1.0.0-beta.7",
	"rollup-plugin-license": "^3.6.0",
	"vitest": "^3.1.1",
	"zx": "^8.5.0"
};
var lint_staged = { "*.{js,ts,vue,json}": ["prettier --write"] };
var publishConfig = {
	"access": "public",
	"provenance": true
};
var package_default = {
	name,
	version,
	description,
	type,
	packageManager,
	bin,
	files,
	engines,
	scripts,
	repository,
	keywords,
	author,
	license,
	bugs,
	homepage,
	devDependencies,
	"lint-staged": lint_staged,
	publishConfig
};

//#endregion
//#region index.ts
const language = await getLanguage(fileURLToPath(new URL("./locales", import.meta.url)));
const FEATURE_FLAGS = [
	"default",
	"ts",
	"typescript",
	"jsx",
	"router",
	"vue-router",
	"pinia",
	"tests",
	"with-tests",
	"vitest",
	"cypress",
	"nightwatch",
	"playwright",
	"eslint",
	"prettier",
	"eslint-with-oxlint",
	"eslint-with-prettier"
];
const FEATURE_OPTIONS = [
	{
		value: "typescript",
		label: language.needsTypeScript.message
	},
	{
		value: "jsx",
		label: language.needsJsx.message
	},
	{
		value: "router",
		label: language.needsRouter.message
	},
	{
		value: "pinia",
		label: language.needsPinia.message
	},
	{
		value: "vitest",
		label: language.needsVitest.message
	},
	{
		value: "e2e",
		label: language.needsE2eTesting.message
	},
	{
		value: "eslint",
		label: language.needsEslint.message
	},
	{
		value: "prettier",
		label: language.needsPrettier.message
	}
];
function isValidPackageName(projectName) {
	return /^(?:@[a-z0-9-*~][a-z0-9-*._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/.test(projectName);
}
function toValidPackageName(projectName) {
	return projectName.trim().toLowerCase().replace(/\s+/g, "-").replace(/^[._]/, "").replace(/[^a-z0-9-~]+/g, "-");
}
function canSkipEmptying(dir) {
	if (!fs.existsSync(dir)) return true;
	const files$1 = fs.readdirSync(dir);
	if (files$1.length === 0) return true;
	if (files$1.length === 1 && files$1[0] === ".git") return true;
	return false;
}
function emptyDir(dir) {
	if (!fs.existsSync(dir)) return;
	postOrderDirectoryTraverse(dir, (dir$1) => fs.rmdirSync(dir$1), (file) => fs.unlinkSync(file));
}
async function unwrapPrompt(maybeCancelPromise) {
	const result = await maybeCancelPromise;
	if (BD(result)) {
		xe((0, import_picocolors.red)("✖") + ` ${language.errors.operationCancelled}`);
		process.exit(0);
	}
	return result;
}
const helpMessage = `\
Usage: create-vue [FEATURE_FLAGS...] [OPTIONS...] [DIRECTORY]

Create a new Vue.js project.
Start the CLI in interactive mode when no FEATURE_FLAGS is provided, or if the DIRECTORY argument is not a valid package name.

Options:
  --force
    Create the project even if the directory is not empty.
  --bare
    Create a barebone project without example code.
  --help
    Display this help message.
  --version
    Display the version number of this CLI.

Available feature flags:
  --default
    Create a project with the default configuration without any additional features.
  --ts, --typescript
    Add TypeScript support.
  --jsx
    Add JSX support.
  --router, --vue-router
    Add Vue Router for SPA development.
  --pinia
    Add Pinia for state management.
  --vitest
    Add Vitest for unit testing.
  --cypress
    Add Cypress for end-to-end testing.
    If used without ${(0, import_picocolors.cyan)("--vitest")}, it will also add Cypress Component Testing.
  --playwright
    Add Playwright for end-to-end testing.
  --nightwatch
    Add Nightwatch for end-to-end testing.
    If used without ${(0, import_picocolors.cyan)("--vitest")}, it will also add Nightwatch Component Testing.
  --eslint
    Add ESLint for code quality.
  --eslint-with-oxlint
    Add ESLint for code quality, and use Oxlint to speed up the linting process.
  --eslint-with-prettier (Deprecated in favor of ${(0, import_picocolors.cyan)("--eslint --prettier")})
    Add Prettier for code formatting in addition to ESLint.
  --prettier
    Add Prettier for code formatting.

Unstable feature flags:
  --tests, --with-tests
    Add both unit testing and end-to-end testing support.
    Currently equivalent to ${(0, import_picocolors.cyan)("--vitest --cypress")}, but may change in the future.
`;
async function init() {
	const cwd = process.cwd();
	const args = process.argv.slice(2);
	const flags = [
		...FEATURE_FLAGS,
		"force",
		"bare",
		"help",
		"version"
	];
	const options = Object.fromEntries(flags.map((key) => [key, { type: "boolean" }]));
	const { values: argv$1, positionals } = parseArgs({
		args,
		options,
		strict: true,
		allowPositionals: true
	});
	if (argv$1.help) {
		console.log(helpMessage);
		process.exit(0);
	}
	if (argv$1.version) {
		console.log(`${package_default.name} v${package_default.version}`);
		process.exit(0);
	}
	const isFeatureFlagsUsed = FEATURE_FLAGS.some((flag) => typeof argv$1[flag] === "boolean");
	let targetDir = positionals[0];
	const defaultProjectName = targetDir || "vue-project";
	const forceOverwrite = argv$1.force;
	const result = {
		projectName: defaultProjectName,
		shouldOverwrite: forceOverwrite,
		packageName: defaultProjectName,
		features: [],
		e2eFramework: void 0,
		experimentOxlint: false
	};
	Ie(process.stdout.isTTY && process.stdout.getColorDepth() > 8 ? gradientBanner : defaultBanner);
	if (!targetDir) {
		const _result = await unwrapPrompt(he({
			message: language.projectName.message,
			placeholder: defaultProjectName,
			validate: (value) => value.trim().length > 0 ? void 0 : language.projectName.invalidMessage
		}));
		targetDir = result.projectName = result.packageName = _result.trim();
	}
	if (!canSkipEmptying(targetDir) && !forceOverwrite) {
		result.shouldOverwrite = await unwrapPrompt(ye({
			message: `${targetDir === "." ? language.shouldOverwrite.dirForPrompts.current : `${language.shouldOverwrite.dirForPrompts.target} "${targetDir}"`} ${language.shouldOverwrite.message}`,
			initialValue: false
		}));
		if (!result.shouldOverwrite) {
			xe((0, import_picocolors.red)("✖") + ` ${language.errors.operationCancelled}`);
			process.exit(0);
		}
	}
	if (!isValidPackageName(targetDir)) result.packageName = await unwrapPrompt(he({
		message: language.packageName.message,
		initialValue: toValidPackageName(targetDir),
		validate: (value) => isValidPackageName(value) ? void 0 : language.packageName.invalidMessage
	}));
	if (!isFeatureFlagsUsed) {
		result.features = await unwrapPrompt(fe({
			message: `${language.featureSelection.message} ${(0, import_picocolors.dim)(language.featureSelection.hint)}`,
			options: FEATURE_OPTIONS,
			required: false
		}));
		if (result.features.includes("e2e")) {
			const hasVitest = result.features.includes("vitest");
			result.e2eFramework = await unwrapPrompt(ve({
				message: `${language.e2eSelection.message} ${(0, import_picocolors.dim)(language.e2eSelection.hint)}`,
				options: [
					{
						value: "playwright",
						label: language.e2eSelection.selectOptions.playwright.title,
						hint: language.e2eSelection.selectOptions.playwright.desc
					},
					{
						value: "cypress",
						label: language.e2eSelection.selectOptions.cypress.title,
						hint: hasVitest ? language.e2eSelection.selectOptions.cypress.desc : language.e2eSelection.selectOptions.cypress.hintOnComponentTesting
					},
					{
						value: "nightwatch",
						label: language.e2eSelection.selectOptions.nightwatch.title,
						hint: hasVitest ? language.e2eSelection.selectOptions.nightwatch.desc : language.e2eSelection.selectOptions.nightwatch.hintOnComponentTesting
					}
				]
			}));
		}
		if (result.features.includes("eslint")) result.experimentOxlint = await unwrapPrompt(ye({
			message: language.needsOxlint.message,
			initialValue: false
		}));
	}
	const { features } = result;
	const needsTypeScript = argv$1.ts || argv$1.typescript || features.includes("typescript");
	const needsJsx = argv$1.jsx || features.includes("jsx");
	const needsRouter = argv$1.router || argv$1["vue-router"] || features.includes("router");
	const needsPinia = argv$1.pinia || features.includes("pinia");
	const needsVitest = argv$1.vitest || argv$1.tests || features.includes("vitest");
	const needsEslint = argv$1.eslint || argv$1["eslint-with-oxlint"] || argv$1["eslint-with-prettier"] || features.includes("eslint");
	const needsPrettier = argv$1.prettier || argv$1["eslint-with-prettier"] || features.includes("prettier");
	const needsOxlint = argv$1["eslint-with-oxlint"] || result.experimentOxlint;
	const { e2eFramework } = result;
	const needsCypress = argv$1.cypress || argv$1.tests || e2eFramework === "cypress";
	const needsCypressCT = needsCypress && !needsVitest;
	const needsNightwatch = argv$1.nightwatch || e2eFramework === "nightwatch";
	const needsNightwatchCT = needsNightwatch && !needsVitest;
	const needsPlaywright = argv$1.playwright || e2eFramework === "playwright";
	const root = path.join(cwd, targetDir);
	if (fs.existsSync(root) && result.shouldOverwrite) emptyDir(root);
	else if (!fs.existsSync(root)) fs.mkdirSync(root);
	console.log(`\n${language.infos.scaffolding} ${root}...`);
	const pkg = {
		name: result.packageName,
		version: "0.0.0"
	};
	fs.writeFileSync(path.resolve(root, "package.json"), JSON.stringify(pkg, null, 2));
	const templateRoot = fileURLToPath(new URL("./template", import.meta.url));
	const callbacks = [];
	const render = function render$1(templateName) {
		const templateDir = path.resolve(templateRoot, templateName);
		renderTemplate_default(templateDir, root, callbacks);
	};
	render("base");
	if (needsJsx) render("config/jsx");
	if (needsRouter) render("config/router");
	if (needsPinia) render("config/pinia");
	if (needsVitest) render("config/vitest");
	if (needsCypress) render("config/cypress");
	if (needsCypressCT) render("config/cypress-ct");
	if (needsNightwatch) render("config/nightwatch");
	if (needsNightwatchCT) render("config/nightwatch-ct");
	if (needsPlaywright) render("config/playwright");
	if (needsTypeScript) {
		render("config/typescript");
		render("tsconfig/base");
		const rootTsConfig = {
			files: [],
			references: [{ path: "./tsconfig.node.json" }, { path: "./tsconfig.app.json" }]
		};
		if (needsCypress) {
			render("tsconfig/cypress");
			rootTsConfig.compilerOptions = { module: "NodeNext" };
		}
		if (needsCypressCT) {
			render("tsconfig/cypress-ct");
			rootTsConfig.references.push({ path: "./tsconfig.cypress-ct.json" });
		}
		if (needsPlaywright) render("tsconfig/playwright");
		if (needsVitest) {
			render("tsconfig/vitest");
			rootTsConfig.references.push({ path: "./tsconfig.vitest.json" });
		}
		if (needsNightwatch) {
			render("tsconfig/nightwatch");
			rootTsConfig.references.push({ path: "./nightwatch/tsconfig.json" });
		}
		if (needsNightwatchCT) render("tsconfig/nightwatch-ct");
		fs.writeFileSync(path.resolve(root, "tsconfig.json"), JSON.stringify(rootTsConfig, null, 2) + "\n", "utf-8");
	}
	if (needsEslint) {
		renderEslint(root, {
			needsTypeScript,
			needsOxlint,
			needsVitest,
			needsCypress,
			needsCypressCT,
			needsPrettier,
			needsPlaywright
		});
		render("config/eslint");
	}
	if (needsOxlint) render("config/oxlint");
	if (needsPrettier) render("config/prettier");
	const codeTemplate = (needsTypeScript ? "typescript-" : "") + (needsRouter ? "router" : "default");
	render(`code/${codeTemplate}`);
	if (needsPinia && needsRouter) render("entry/router-and-pinia");
	else if (needsPinia) render("entry/pinia");
	else if (needsRouter) render("entry/router");
	else render("entry/default");
	const dataStore = {};
	for (const cb of callbacks) await cb(dataStore);
	preOrderDirectoryTraverse(root, () => {}, (filepath) => {
		if (filepath.endsWith(".ejs")) {
			const template = fs.readFileSync(filepath, "utf-8");
			const dest = filepath.replace(/\.ejs$/, "");
			const content = import_ejs.render(template, dataStore[dest]);
			fs.writeFileSync(dest, content);
			fs.unlinkSync(filepath);
		}
	});
	if (argv$1.bare) {
		trimBoilerplate(root);
		render("bare/base");
		if (needsTypeScript) render("bare/typescript");
		if (needsVitest) render("bare/vitest");
		if (needsCypressCT) render("bare/cypress-ct");
		if (needsNightwatchCT) render("bare/nightwatch-ct");
	}
	if (needsTypeScript) {
		preOrderDirectoryTraverse(root, () => {}, (filepath) => {
			if (filepath.endsWith(".js") && !filepath.endsWith("eslint.config.js")) {
				const tsFilePath = filepath.replace(/\.js$/, ".ts");
				if (fs.existsSync(tsFilePath)) fs.unlinkSync(filepath);
				else fs.renameSync(filepath, tsFilePath);
			} else if (path.basename(filepath) === "jsconfig.json") fs.unlinkSync(filepath);
		});
		const indexHtmlPath = path.resolve(root, "index.html");
		const indexHtmlContent = fs.readFileSync(indexHtmlPath, "utf8");
		fs.writeFileSync(indexHtmlPath, indexHtmlContent.replace("src/main.js", "src/main.ts"));
	} else preOrderDirectoryTraverse(root, () => {}, (filepath) => {
		if (filepath.endsWith(".ts")) fs.unlinkSync(filepath);
	});
	if (argv$1.bare) {
		removeCSSImport(root, needsTypeScript, needsCypressCT);
		if (needsRouter) emptyRouterConfig(root, needsTypeScript);
	}
	const userAgent = process.env.npm_config_user_agent ?? "";
	const packageManager$1 = /pnpm/.test(userAgent) ? "pnpm" : /yarn/.test(userAgent) ? "yarn" : /bun/.test(userAgent) ? "bun" : "npm";
	fs.writeFileSync(path.resolve(root, "README.md"), generateReadme({
		projectName: result.projectName ?? result.packageName ?? defaultProjectName,
		packageManager: packageManager$1,
		needsTypeScript,
		needsVitest,
		needsCypress,
		needsNightwatch,
		needsPlaywright,
		needsNightwatchCT,
		needsCypressCT,
		needsEslint
	}));
	let outroMessage = `${language.infos.done}\n\n`;
	if (root !== cwd) {
		const cdProjectName = path.relative(cwd, root);
		outroMessage += `   ${(0, import_picocolors.bold)((0, import_picocolors.green)(`cd ${cdProjectName.includes(" ") ? `"${cdProjectName}"` : cdProjectName}`))}\n`;
	}
	outroMessage += `   ${(0, import_picocolors.bold)((0, import_picocolors.green)(getCommand(packageManager$1, "install")))}\n`;
	if (needsPrettier) outroMessage += `   ${(0, import_picocolors.bold)((0, import_picocolors.green)(getCommand(packageManager$1, "format")))}\n`;
	outroMessage += `   ${(0, import_picocolors.bold)((0, import_picocolors.green)(getCommand(packageManager$1, "dev")))}\n`;
	outroMessage += `
${(0, import_picocolors.dim)("|")} ${language.infos.optionalGitCommand}
   
   ${(0, import_picocolors.bold)((0, import_picocolors.green)("git init && git add -A && git commit -m \"initial commit\""))}`;
	Se(outroMessage);
}
init().catch((e) => {
	console.error(e);
	process.exit(1);
});

//#endregion
